Index: llvm-toolchain-12_12.0.0~++20210204121419+e8cdcaeae406/openmp/runtime/src/CMakeLists.txt
===================================================================
--- llvm-toolchain-12_12.0.0~++20210204121419+e8cdcaeae406.orig/openmp/runtime/src/CMakeLists.txt
+++ llvm-toolchain-12_12.0.0~++20210204121419+e8cdcaeae406/openmp/runtime/src/CMakeLists.txt
@@ -146,7 +146,8 @@ libomp_get_libflags(LIBOMP_CONFIGURED_LI
 if(OPENMP_STANDALONE_BUILD OR (NOT OPENMP_ENABLE_LIBOMP_PROFILING))
   add_library(omp ${LIBOMP_LIBRARY_KIND} ${LIBOMP_SOURCE_FILES})
   # Linking command will include libraries in LIBOMP_CONFIGURED_LIBFLAGS
-  target_link_libraries(omp ${LIBOMP_CONFIGURED_LIBFLAGS} ${CMAKE_DL_LIBS})
+  target_link_libraries(omp ${LIBOMP_CONFIGURED_LIBFLAGS} ${CMAKE_DL_LIBS}
+    "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/exports_so.txt")
 else()
   add_llvm_library(omp ${LIBOMP_LIBRARY_KIND} ${LIBOMP_SOURCE_FILES} PARTIAL_SOURCES_INTENDED
     LINK_LIBS ${LIBOMP_CONFIGURED_LIBFLAGS} ${CMAKE_DL_LIBS}
