Index: llvm-toolchain-snapshot_9~svn351647/lldb/tools/argdumper/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_9~svn351647.orig/lldb/tools/argdumper/CMakeLists.txt
+++ llvm-toolchain-snapshot_9~svn351647/lldb/tools/argdumper/CMakeLists.txt
@@ -4,3 +4,7 @@ add_lldb_tool(lldb-argdumper
   LINK_LIBS
     lldbUtility
   )
+
+install(TARGETS lldb-argdumper
+  RUNTIME DESTINATION bin)
+
Index: llvm-toolchain-snapshot_9~svn351647/lldb/tools/lldb-server/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_9~svn351647.orig/lldb/tools/lldb-server/CMakeLists.txt
+++ llvm-toolchain-snapshot_9~svn351647/lldb/tools/lldb-server/CMakeLists.txt
@@ -64,3 +64,7 @@ add_lldb_tool(lldb-server
 )
 
 target_link_libraries(lldb-server PRIVATE ${LLDB_SYSTEM_LIBS})
+
+install(TARGETS lldb-server
+  RUNTIME DESTINATION bin)
+
