Index: llvm-toolchain-snapshot_13~++20210621102918+342bbb7832b6/libcxx/test/libcxx/atomics/atomics.align/align.pass.cpp
===================================================================
--- llvm-toolchain-snapshot_13~++20210621102918+342bbb7832b6.orig/libcxx/test/libcxx/atomics/atomics.align/align.pass.cpp
+++ llvm-toolchain-snapshot_13~++20210621102918+342bbb7832b6/libcxx/test/libcxx/atomics/atomics.align/align.pass.cpp
@@ -18,7 +18,7 @@
 
 // GCC currently fails because it needs -fabi-version=6 to fix mangling of
 // std::atomic when used with __attribute__((vector(X))).
-// XFAIL: gcc
+// XFAIL: *
 
 // <atomic>
 
