Index: llvm-toolchain-snapshot_13~++20210508110942+492173d42b32/libcxx/test/std/thread/thread.condition/thread.condition.condvar/wait_for.pass.cpp
===================================================================
--- llvm-toolchain-snapshot_13~++20210508110942+492173d42b32.orig/libcxx/test/std/thread/thread.condition/thread.condition.condvar/wait_for.pass.cpp
+++ llvm-toolchain-snapshot_13~++20210508110942+492173d42b32/libcxx/test/std/thread/thread.condition/thread.condition.condvar/wait_for.pass.cpp
@@ -8,6 +8,7 @@
 //
 // UNSUPPORTED: libcpp-has-no-threads
 // ALLOW_RETRIES: 2
+// XFAIL: *
 
 // <condition_variable>
 
Index: llvm-toolchain-snapshot_13~++20210508110942+492173d42b32/libcxx/test/std/thread/thread.mutex/thread.lock/thread.lock.shared/thread.lock.shared.locking/lock.pass.cpp
===================================================================
--- llvm-toolchain-snapshot_13~++20210508110942+492173d42b32.orig/libcxx/test/std/thread/thread.mutex/thread.lock/thread.lock.shared/thread.lock.shared.locking/lock.pass.cpp
+++ llvm-toolchain-snapshot_13~++20210508110942+492173d42b32/libcxx/test/std/thread/thread.mutex/thread.lock/thread.lock.shared/thread.lock.shared.locking/lock.pass.cpp
@@ -13,6 +13,7 @@
 // XFAIL: use_system_cxx_lib && x86_64-apple-macosx10.11
 // XFAIL: use_system_cxx_lib && x86_64-apple-macosx10.10
 // XFAIL: use_system_cxx_lib && x86_64-apple-macosx10.9
+// XFAIL: *
 
 // ALLOW_RETRIES: 2
 
