Index: llvm-toolchain-snapshot_3.4~svn183420/lib/Support/CommandLine.cpp
===================================================================
--- llvm-toolchain-snapshot_3.4~svn183420.orig/lib/Support/CommandLine.cpp	2013-05-10 19:15:51.000000000 +0200
+++ llvm-toolchain-snapshot_3.4~svn183420/lib/Support/CommandLine.cpp	2013-06-06 18:37:07.000000000 +0200
@@ -1521,6 +1521,10 @@
     OS << LLVM_VERSION_INFO;
 #endif
     OS << "\n  ";
+#ifdef LLVM_DEBIAN_INFO
+    OS << LLVM_DEBIAN_INFO;
+#endif
+    OS << "\n  ";
 #ifndef __OPTIMIZE__
     OS << "DEBUG build";
 #else
