Index: llvm-toolchain-snapshot_3.4~svn188146/lib/Support/CommandLine.cpp
===================================================================
--- llvm-toolchain-snapshot_3.4~svn188146.orig/lib/Support/CommandLine.cpp	2013-08-11 10:01:30.736734807 +0200
+++ llvm-toolchain-snapshot_3.4~svn188146/lib/Support/CommandLine.cpp	2013-08-11 10:01:30.732734807 +0200
@@ -1686,6 +1686,10 @@
     OS << LLVM_VERSION_INFO;
 #endif
     OS << "\n  ";
+#ifdef LLVM_DEBIAN_INFO
+    OS << LLVM_DEBIAN_INFO;
+#endif
+    OS << "\n  ";
 #ifndef __OPTIMIZE__
     OS << "DEBUG build";
 #else
