Index: llvm-toolchain-build/lib/Support/CommandLine.cpp
===================================================================
--- llvm-toolchain-build.orig/lib/Support/CommandLine.cpp	2013-03-15 09:40:15.000000000 +0100
+++ llvm-toolchain-build/lib/Support/CommandLine.cpp	2013-03-15 09:43:09.000000000 +0100
@@ -1347,6 +1347,10 @@
     OS << LLVM_VERSION_INFO;
 #endif
     OS << "\n  ";
+#ifdef LLVM_DEBIAN_INFO
+    OS << LLVM_DEBIAN_INFO;
+#endif
+    OS << "\n  ";
 #ifndef __OPTIMIZE__
     OS << "DEBUG build";
 #else
