Index: llvm-toolchain-snapshot_3.5~svn200375/lib/Support/CommandLine.cpp
===================================================================
--- llvm-toolchain-snapshot_3.5~svn200375.orig/lib/Support/CommandLine.cpp	2014-01-29 07:45:46.092720538 -0800
+++ llvm-toolchain-snapshot_3.5~svn200375/lib/Support/CommandLine.cpp	2014-01-29 07:45:46.092720538 -0800
@@ -1691,6 +1691,10 @@
     OS << LLVM_VERSION_INFO;
 #endif
     OS << "\n  ";
+#ifdef LLVM_DEBIAN_INFO
+    OS << LLVM_DEBIAN_INFO;
+#endif
+    OS << "\n  ";
 #ifndef __OPTIMIZE__
     OS << "DEBUG build";
 #else
