Index: llvm-toolchain_3.3~svn176561/tools/llvm-shlib/Makefile
===================================================================
--- llvm-toolchain_3.3~svn176561.orig/tools/llvm-shlib/Makefile	2013-03-06 18:51:25.528715188 +0100
+++ llvm-toolchain_3.3~svn176561/tools/llvm-shlib/Makefile	2013-03-06 18:51:27.032715158 +0100
@@ -77,7 +77,7 @@
 
 ifeq ($(HOST_OS), $(filter $(HOST_OS), Linux FreeBSD GNU))
     # Add soname to the library.
-    LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME)$(SHLIBEXT) -Wl,-Bsymbolic-functions
+    LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME)$(SHLIBEXT).1 -Wl,-Bsymbolic-functions
 endif
 
 ifeq ($(HOST_OS), $(filter $(HOST_OS), Linux GNU))
