Index: llvm-toolchain-snapshot_3.4~svn183420/tools/llvm-shlib/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.4~svn183420.orig/tools/llvm-shlib/Makefile	2013-06-06 18:37:08.000000000 +0200
+++ llvm-toolchain-snapshot_3.4~svn183420/tools/llvm-shlib/Makefile	2013-06-06 18:37:09.000000000 +0200
@@ -76,7 +76,7 @@
 
 ifeq ($(HOST_OS), $(filter $(HOST_OS), Linux FreeBSD GNU))
     # Add soname to the library.
-    LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME)$(SHLIBEXT) -Wl,-Bsymbolic-functions
+    LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME)$(SHLIBEXT).1 -Wl,-Bsymbolic-functions
 endif
 
 ifeq ($(HOST_OS), $(filter $(HOST_OS), Linux GNU))
