Index: llvm-toolchain-3.7_3.7.1~+rc1/tools/llvm-shlib/Makefile
===================================================================
--- llvm-toolchain-3.7_3.7.1~+rc1.orig/tools/llvm-shlib/Makefile
+++ llvm-toolchain-3.7_3.7.1~+rc1/tools/llvm-shlib/Makefile
@@ -67,7 +67,7 @@ endif
 
 ifeq ($(HOST_OS), $(filter $(HOST_OS), DragonFly Linux FreeBSD GNU/kFreeBSD GNU))
     # Add soname to the library.
-    LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME).1$(SHLIBEXT) -Wl,-Bsymbolic-functions
+    LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME)$(SHLIBEXT).1 -Wl,-Bsymbolic-functions
 endif
 
 ifeq ($(HOST_OS), $(filter $(HOST_OS), Linux GNU GNU/kFreeBSD))
