Index: llvm-toolchain-snapshot_3.4~svn188146/tools/llvm-shlib/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.4~svn188146.orig/tools/llvm-shlib/Makefile	2013-08-11 10:01:32.084734772 +0200
+++ llvm-toolchain-snapshot_3.4~svn188146/tools/llvm-shlib/Makefile	2013-08-11 10:01:32.080734772 +0200
@@ -76,7 +76,7 @@
 
 ifeq ($(HOST_OS), $(filter $(HOST_OS), Linux FreeBSD GNU/kFreeBSD GNU))
     # Add soname to the library.
-    LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME)$(SHLIBEXT) -Wl,-Bsymbolic-functions
+    LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME)$(SHLIBEXT).1 -Wl,-Bsymbolic-functions
 endif
 
 ifeq ($(HOST_OS), $(filter $(HOST_OS), Linux GNU GNU/kFreeBSD))
