Index: llvm-toolchain-snapshot_3.5~svn201068/autoconf/configure.ac
===================================================================
--- llvm-toolchain-snapshot_3.5~svn201068.orig/autoconf/configure.ac	2014-02-10 09:12:31.462814967 +0100
+++ llvm-toolchain-snapshot_3.5~svn201068/autoconf/configure.ac	2014-02-10 09:12:49.866814516 +0100
@@ -31,7 +31,7 @@
 dnl===-----------------------------------------------------------------------===
 dnl Initialize autoconf and define the package name, version number and
 dnl address for reporting bugs.
-AC_INIT([LLVM],[3.5svn],[http://llvm.org/bugs/])
+AC_INIT([LLVM],[3.5],[http://llvm.org/bugs/])
 LLVM_DEFINE_SUBST([LLVM_VERSION_MAJOR], [3], [Major version of the LLVM API])
 LLVM_DEFINE_SUBST([LLVM_VERSION_MINOR], [5], [Minor version of the LLVM API])
 
Index: llvm-toolchain-snapshot_3.5~svn201068/configure
===================================================================
--- llvm-toolchain-snapshot_3.5~svn201068.orig/configure	2014-02-10 09:12:31.462814967 +0100
+++ llvm-toolchain-snapshot_3.5~svn201068/configure	2014-02-10 09:12:31.462814967 +0100
@@ -561,8 +561,8 @@
 # Identity of this package.
 PACKAGE_NAME='LLVM'
 PACKAGE_TARNAME='llvm'
-PACKAGE_VERSION='3.5svn'
-PACKAGE_STRING='LLVM 3.5svn'
+PACKAGE_VERSION='3.5'
+PACKAGE_STRING='LLVM 3.5'
 PACKAGE_BUGREPORT='http://llvm.org/bugs/'
 
 ac_unique_file="lib/IR/Module.cpp"
