Index: llvm-toolchain-snapshot_3.6~svn214390/autoconf/configure.ac
===================================================================
--- llvm-toolchain-snapshot_3.6~svn214390.orig/autoconf/configure.ac
+++ llvm-toolchain-snapshot_3.6~svn214390/autoconf/configure.ac
@@ -32,7 +32,7 @@ dnl===----------------------------------
 dnl Initialize autoconf and define the package name, version number and
 dnl address for reporting bugs.
 
-AC_INIT([LLVM],[3.6.0svn],[http://llvm.org/bugs/])
+AC_INIT([LLVM],[3.6.0],[http://llvm.org/bugs/])
 
 LLVM_VERSION_MAJOR=3
 LLVM_VERSION_MINOR=6
Index: llvm-toolchain-snapshot_3.6~svn214390/configure
===================================================================
--- llvm-toolchain-snapshot_3.6~svn214390.orig/configure
+++ llvm-toolchain-snapshot_3.6~svn214390/configure
@@ -561,8 +561,8 @@ SHELL=${CONFIG_SHELL-/bin/sh}
 # Identity of this package.
 PACKAGE_NAME='LLVM'
 PACKAGE_TARNAME='llvm'
-PACKAGE_VERSION='3.6.0svn'
-PACKAGE_STRING='LLVM 3.6.0svn'
+PACKAGE_VERSION='3.6.0'
+PACKAGE_STRING='LLVM 3.6.0'
 PACKAGE_BUGREPORT='http://llvm.org/bugs/'
 
 ac_unique_file="lib/IR/Module.cpp"
