Index: llvm-toolchain-snapshot_3.4~svn185325/clang/tools/libclang/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.4~svn185325.orig/clang/tools/libclang/Makefile	2013-07-01 13:26:38.000000000 +0200
+++ llvm-toolchain-snapshot_3.4~svn185325/clang/tools/libclang/Makefile	2013-07-01 13:27:07.000000000 +0200
@@ -14,6 +14,7 @@
 
 LINK_LIBS_IN_SHARED = 1
 SHARED_LIBRARY = 1
+SONAME_MAJOR=1
 
 include $(CLANG_LEVEL)/../../Makefile.config
 LINK_COMPONENTS := $(TARGETS_TO_BUILD) asmparser bitreader support mc option
@@ -30,7 +31,7 @@
 
 # Add soname to the library.
 ifeq ($(HOST_OS), $(filter $(HOST_OS), Linux FreeBSD GNU GNU/kFreeBSD))
-        LDFLAGS += -Wl,-soname,lib$(LIBRARYNAME)$(SHLIBEXT)
+        LDFLAGS += -Wl,-soname,lib$(LIBRARYNAME)$(SHLIBEXT).$(SONAME_MAJOR)
 endif
 
 ##===----------------------------------------------------------------------===##
