Index: llvm-toolchain-snapshot_10~+201911111222550800646d927175e/clang/lib/Basic/Version.cpp
===================================================================
--- llvm-toolchain-snapshot_10~+201911111222550800646d927175e.orig/clang/lib/Basic/Version.cpp
+++ llvm-toolchain-snapshot_10~+201911111222550800646d927175e/clang/lib/Basic/Version.cpp
@@ -14,6 +14,7 @@
 #include "clang/Basic/LLVM.h"
 #include "clang/Config/config.h"
 #include "llvm/Support/raw_ostream.h"
+#include "clang/Debian/debian_path.h"
 #include <cstdlib>
 #include <cstring>
 
@@ -124,7 +125,7 @@ std::string getClangToolFullVersion(Stri
 #ifdef CLANG_VENDOR
   OS << CLANG_VENDOR;
 #endif
-  OS << ToolName << " version " CLANG_VERSION_STRING " "
+  OS << ToolName << " version " CLANG_VERSION_STRING "-" DEB_PATCHSETVERSION " "
      << getClangFullRepositoryVersion();
 
   return OS.str();
