Index: llvm-toolchain-snapshot_11~++20200708031542+64363a9d930/clang/lib/Basic/Version.cpp
===================================================================
--- llvm-toolchain-snapshot_11~++20200708031542+64363a9d930.orig/clang/lib/Basic/Version.cpp
+++ llvm-toolchain-snapshot_11~++20200708031542+64363a9d930/clang/lib/Basic/Version.cpp
@@ -14,6 +14,7 @@
 #include "clang/Basic/LLVM.h"
 #include "clang/Config/config.h"
 #include "llvm/Support/raw_ostream.h"
+#include "clang/Debian/debian_path.h"
 #include <cstdlib>
 #include <cstring>
 
@@ -97,7 +98,7 @@ std::string getClangToolFullVersion(Stri
 #ifdef CLANG_VENDOR
   OS << CLANG_VENDOR;
 #endif
-  OS << ToolName << " version " CLANG_VERSION_STRING;
+  OS << ToolName << " version " CLANG_VERSION_STRING "-" DEB_PATCHSETVERSION;
 
   std::string repo = getClangFullRepositoryVersion();
   if (!repo.empty()) {
