Index: llvm-toolchain_3.3~svn179457/clang/Makefile
===================================================================
--- llvm-toolchain_3.3~svn179457.orig/clang/Makefile	2012-10-03 10:39:19.000000000 +0200
+++ llvm-toolchain_3.3~svn179457/clang/Makefile	2013-04-13 10:29:14.857932989 +0200
@@ -47,9 +47,7 @@
 
 # Set common Clang build flags.
 CPP.Flags += -I$(PROJ_SRC_DIR)/$(CLANG_LEVEL)/include -I$(PROJ_OBJ_DIR)/$(CLANG_LEVEL)/include
-ifdef CLANG_VENDOR
-CPP.Flags += -DCLANG_VENDOR='"$(CLANG_VENDOR) "'
-endif
+CPP.Flags += -DCLANG_VENDOR='"Debian "'
 ifdef CLANG_REPOSITORY_STRING
 CPP.Flags += -DCLANG_REPOSITORY_STRING='"$(CLANG_REPOSITORY_STRING)"'
 endif
