Index: llvm-toolchain-snapshot_3.4~svn184290/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain-snapshot_3.4~svn184290.orig/clang/lib/Driver/ToolChains.cpp	2013-06-19 11:13:55.238779496 +0200
+++ llvm-toolchain-snapshot_3.4~svn184290/clang/lib/Driver/ToolChains.cpp	2013-06-19 11:13:55.234779499 +0200
@@ -2409,6 +2409,11 @@
     addSystemInclude(DriverArgs, CC1Args, P.str());
   }
 
+  // Under Debian, clang headers are installed into
+  // '/usr/include/clang/VERSION/include/'
+  addSystemInclude(DriverArgs, CC1Args, "/usr/include/clang/" + std::string(CLANG_VERSION_STRING) + "/include/");
+  
+
   if (DriverArgs.hasArg(options::OPT_nostdlibinc))
     return;
 
