Index: llvm-toolchain-snapshot_3.4~svn185325/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain-snapshot_3.4~svn185325.orig/clang/lib/Driver/ToolChains.cpp	2013-07-01 13:27:19.000000000 +0200
+++ llvm-toolchain-snapshot_3.4~svn185325/clang/lib/Driver/ToolChains.cpp	2013-07-01 13:27:19.000000000 +0200
@@ -2366,6 +2366,11 @@
     addSystemInclude(DriverArgs, CC1Args, P.str());
   }
 
+  // Under Debian, clang headers are installed into
+  // '/usr/include/clang/VERSION/include/'
+  addSystemInclude(DriverArgs, CC1Args, "/usr/include/clang/" + std::string(CLANG_VERSION_STRING) + "/include/");
+  
+
   if (DriverArgs.hasArg(options::OPT_nostdlibinc))
     return;
 
