Index: llvm-toolchain_3.3~svn176083/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain_3.3~svn176083.orig/clang/lib/Driver/ToolChains.cpp	2013-02-21 23:35:01.000000000 +0100
+++ llvm-toolchain_3.3~svn176083/clang/lib/Driver/ToolChains.cpp	2013-02-26 14:10:01.000000000 +0100
@@ -2487,6 +2487,11 @@
     addSystemInclude(DriverArgs, CC1Args, P.str());
   }
 
+  // Under Debian, clang headers are installed into
+  // '/usr/include/clang/VERSION/include/'
+  addSystemInclude(DriverArgs, CC1Args, "/usr/include/clang/" + std::string(CLANG_VERSION_STRING) + "/include/");
+  
+
   if (DriverArgs.hasArg(options::OPT_nostdlibinc))
     return;
 
