Index: llvm-toolchain-snapshot_3.4~svn188146/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain-snapshot_3.4~svn188146.orig/clang/lib/Driver/ToolChains.cpp	2013-08-11 10:01:27.552734888 +0200
+++ llvm-toolchain-snapshot_3.4~svn188146/clang/lib/Driver/ToolChains.cpp	2013-08-11 10:01:27.548734887 +0200
@@ -2415,6 +2415,11 @@
     addSystemInclude(DriverArgs, CC1Args, P.str());
   }
 
+  // Under Debian, clang headers are installed into
+  // '/usr/include/clang/VERSION/include/'
+  addSystemInclude(DriverArgs, CC1Args, "/usr/include/clang/" + std::string(CLANG_VERSION_STRING) + "/include/");
+  
+
   if (DriverArgs.hasArg(options::OPT_nostdlibinc))
     return;
 
