Index: llvm-toolchain_3.3~svn179457/clang/lib/Lex/HeaderSearch.cpp
===================================================================
--- llvm-toolchain_3.3~svn179457.orig/clang/lib/Lex/HeaderSearch.cpp	2013-03-21 02:08:50.000000000 +0100
+++ llvm-toolchain_3.3~svn179457/clang/lib/Lex/HeaderSearch.cpp	2013-04-13 10:29:33.101932282 +0200
@@ -28,6 +28,11 @@
 #endif
 using namespace clang;
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 const IdentifierInfo *
 HeaderFileInfo::getControllingMacro(ExternalIdentifierLookup *External) {
   if (ControllingMacro)
Index: llvm-toolchain_3.3~svn179457/clang/lib/Lex/ModuleMap.cpp
===================================================================
--- llvm-toolchain_3.3~svn179457.orig/clang/lib/Lex/ModuleMap.cpp	2013-03-20 22:10:35.000000000 +0100
+++ llvm-toolchain_3.3~svn179457/clang/lib/Lex/ModuleMap.cpp	2013-04-13 10:29:33.101932282 +0200
@@ -35,6 +35,11 @@
 #endif
 using namespace clang;
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 Module::ExportDecl 
 ModuleMap::resolveExport(Module *Mod, 
                          const Module::UnresolvedExportDecl &Unresolved,
