Index: llvm-toolchain_3.3~svn176083/clang/lib/Lex/HeaderSearch.cpp
===================================================================
--- llvm-toolchain_3.3~svn176083.orig/clang/lib/Lex/HeaderSearch.cpp	2013-02-08 01:10:48.000000000 +0100
+++ llvm-toolchain_3.3~svn176083/clang/lib/Lex/HeaderSearch.cpp	2013-02-26 14:11:46.000000000 +0100
@@ -28,6 +28,11 @@
 #endif
 using namespace clang;
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 const IdentifierInfo *
 HeaderFileInfo::getControllingMacro(ExternalIdentifierLookup *External) {
   if (ControllingMacro)
Index: llvm-toolchain_3.3~svn176083/clang/lib/Lex/ModuleMap.cpp
===================================================================
--- llvm-toolchain_3.3~svn176083.orig/clang/lib/Lex/ModuleMap.cpp	2013-02-19 20:58:45.000000000 +0100
+++ llvm-toolchain_3.3~svn176083/clang/lib/Lex/ModuleMap.cpp	2013-02-26 14:11:46.000000000 +0100
@@ -34,6 +34,11 @@
 #endif
 using namespace clang;
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 Module::ExportDecl 
 ModuleMap::resolveExport(Module *Mod, 
                          const Module::UnresolvedExportDecl &Unresolved,
