Index: llvm-toolchain-build/clang/lib/Lex/HeaderSearch.cpp
===================================================================
--- llvm-toolchain-build.orig/clang/lib/Lex/HeaderSearch.cpp	2013-03-15 09:40:16.000000000 +0100
+++ llvm-toolchain-build/clang/lib/Lex/HeaderSearch.cpp	2013-03-15 09:43:07.000000000 +0100
@@ -28,6 +28,11 @@
 #endif
 using namespace clang;
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 const IdentifierInfo *
 HeaderFileInfo::getControllingMacro(ExternalIdentifierLookup *External) {
   if (ControllingMacro)
Index: llvm-toolchain-build/clang/lib/Lex/ModuleMap.cpp
===================================================================
--- llvm-toolchain-build.orig/clang/lib/Lex/ModuleMap.cpp	2013-03-15 09:40:16.000000000 +0100
+++ llvm-toolchain-build/clang/lib/Lex/ModuleMap.cpp	2013-03-15 09:43:07.000000000 +0100
@@ -33,6 +33,11 @@
 #endif
 using namespace clang;
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 Module::ExportDecl 
 ModuleMap::resolveExport(Module *Mod, 
                          const Module::UnresolvedExportDecl &Unresolved,
