Index: llvm-toolchain-snapshot_3.4~svn193982/clang/lib/Driver/ToolChains.h
===================================================================
--- llvm-toolchain-snapshot_3.4~svn193982.orig/clang/lib/Driver/ToolChains.h	2013-11-04 09:08:57.592586614 +0100
+++ llvm-toolchain-snapshot_3.4~svn193982/clang/lib/Driver/ToolChains.h	2013-11-04 09:08:57.588586613 +0100
@@ -164,6 +164,8 @@
   virtual bool isPICDefault() const;
   virtual bool isPIEDefault() const;
   virtual bool isPICDefaultForced() const;
+  virtual void AddClangSystemIncludeArgs(const llvm::opt::ArgList &DriverArgs,
+                                        llvm::opt::ArgStringList &CC1Args) const;
 
 protected:
   virtual Tool *getTool(Action::ActionClass AC) const;
Index: llvm-toolchain-snapshot_3.4~svn193982/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain-snapshot_3.4~svn193982.orig/clang/lib/Driver/ToolChains.cpp	2013-11-04 09:08:57.592586614 +0100
+++ llvm-toolchain-snapshot_3.4~svn193982/clang/lib/Driver/ToolChains.cpp	2013-11-04 09:08:57.592586614 +0100
@@ -960,6 +960,13 @@
   return GoodVersion;
 }
 
+void Generic_GCC::AddClangSystemIncludeArgs(const llvm::opt::ArgList &DriverArgs,
+                                      llvm::opt::ArgStringList &CC1Args) const {
+  // Under Debian, clang headers are installed into
+  // '/usr/include/clang/VERSION/include/'
+  addSystemInclude(DriverArgs, CC1Args, "/usr/include/clang/" + std::string(CLANG_VERSION_STRING) + "/include/");
+}
+
 /// \brief Less-than for GCCVersion, implementing a Strict Weak Ordering.
 bool Generic_GCC::GCCVersion::isOlderThan(int RHSMajor, int RHSMinor,
                                           int RHSPatch,
