Index: llvm-toolchain_3.3~svn176083/clang/lib/Driver/ToolChains.h
===================================================================
--- llvm-toolchain_3.3~svn176083.orig/clang/lib/Driver/ToolChains.h	2013-01-31 13:13:10.000000000 +0100
+++ llvm-toolchain_3.3~svn176083/clang/lib/Driver/ToolChains.h	2013-02-26 14:11:47.000000000 +0100
@@ -129,6 +129,8 @@
   virtual bool IsUnwindTablesDefault() const;
   virtual bool isPICDefault() const;
   virtual bool isPICDefaultForced() const;
+  virtual void AddClangSystemIncludeArgs(const ArgList &DriverArgs,
+                                        ArgStringList &CC1Args) const;
 
 protected:
   /// \name ToolChain Implementation Helper Functions
Index: llvm-toolchain_3.3~svn176083/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain_3.3~svn176083.orig/clang/lib/Driver/ToolChains.cpp	2013-02-26 14:11:45.000000000 +0100
+++ llvm-toolchain_3.3~svn176083/clang/lib/Driver/ToolChains.cpp	2013-02-26 14:11:47.000000000 +0100
@@ -963,6 +963,13 @@
   return GoodVersion;
 }
 
+void Generic_GCC::AddClangSystemIncludeArgs(const ArgList &DriverArgs,
+                                      ArgStringList &CC1Args) const {
+  // Under Debian, clang headers are installed into
+  // '/usr/include/clang/VERSION/include/'
+  addSystemInclude(DriverArgs, CC1Args, "/usr/include/clang/" + std::string(CLANG_VERSION_STRING) + "/include/");
+}
+
 /// \brief Less-than for GCCVersion, implementing a Strict Weak Ordering.
 bool Generic_GCC::GCCVersion::operator<(const GCCVersion &RHS) const {
   if (Major != RHS.Major)
