Index: llvm-toolchain-snapshot_3.4~svn184046/clang/lib/Lex/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.4~svn184046.orig/clang/lib/Lex/Makefile	2013-06-15 19:43:18.000000000 +0200
+++ llvm-toolchain-snapshot_3.4~svn184046/clang/lib/Lex/Makefile	2013-06-15 19:43:18.000000000 +0200
@@ -17,8 +17,10 @@
 LIBRARYNAME := clangLex
 
 ifeq ($(ARCH),PowerPC)
+ifneq ($(shell dpkg-architecture -qDEB_HOST_ARCH),powerpcspe)
 CXX.Flags += -maltivec
 endif
+endif
 
 include $(CLANG_LEVEL)/Makefile
 
