Index: llvm-toolchain-snapshot_3.4~svn184290/clang/lib/Lex/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.4~svn184290.orig/clang/lib/Lex/Makefile	2013-06-19 11:14:01.434779657 +0200
+++ llvm-toolchain-snapshot_3.4~svn184290/clang/lib/Lex/Makefile	2013-06-19 11:14:01.430779659 +0200
@@ -17,8 +17,10 @@
 LIBRARYNAME := clangLex
 
 ifeq ($(ARCH),PowerPC)
+ifneq ($(shell dpkg-architecture -qDEB_HOST_ARCH),powerpcspe)
 CXX.Flags += -maltivec
 endif
+endif
 
 include $(CLANG_LEVEL)/Makefile
 
