Index: llvm-toolchain-snapshot_16~++20221109021920+0f4f246783fa/llvm/lib/Support/ARMTargetParser.cpp
===================================================================
--- llvm-toolchain-snapshot_16~++20221109021920+0f4f246783fa.orig/llvm/lib/Support/ARMTargetParser.cpp
+++ llvm-toolchain-snapshot_16~++20221109021920+0f4f246783fa/llvm/lib/Support/ARMTargetParser.cpp
@@ -704,7 +704,7 @@ StringRef ARM::getARMCPUForArch(const ll
     case llvm::Triple::EABIHF:
     case llvm::Triple::GNUEABIHF:
     case llvm::Triple::MuslEABIHF:
-      return "arm1176jzf-s";
+      return "cortex-a7";
     default:
       return "arm7tdmi";
     }
