Index: llvm-toolchain-9_9.0.0/llvm/lib/Support/Triple.cpp
===================================================================
--- llvm-toolchain-9_9.0.0.orig/llvm/lib/Support/Triple.cpp
+++ llvm-toolchain-9_9.0.0/llvm/lib/Support/Triple.cpp
@@ -1646,7 +1646,7 @@ StringRef Triple::getARMCPUForArch(Strin
     case llvm::Triple::EABIHF:
     case llvm::Triple::GNUEABIHF:
     case llvm::Triple::MuslEABIHF:
-      return "arm1176jzf-s";
+      return "cortex-a7";
     default:
       return "arm7tdmi";
     }
