Index: llvm-toolchain-9-9~+rc1/libcxx/include/__locale
===================================================================
--- llvm-toolchain-9-9~+rc1.orig/libcxx/include/__locale
+++ llvm-toolchain-9-9~+rc1/libcxx/include/__locale
@@ -409,7 +409,7 @@ public:
     static const mask xdigit = _ISxdigit;
     static const mask blank  = _ISblank;
 #if defined(__mips__)
-    static const mask __regex_word = static_cast<char_class_type>(_ISbit(15));
+    static const mask __regex_word = static_cast<mask>(_ISbit(15));
 #else
     static const mask __regex_word = 0x80;
 #endif
