Index: llvm-toolchain-snapshot_11~++20200518084103+b7924d6525b/clang-tools-extra/clangd/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_11~++20200518084103+b7924d6525b.orig/clang-tools-extra/clangd/CMakeLists.txt
+++ llvm-toolchain-snapshot_11~++20200518084103+b7924d6525b/clang-tools-extra/clangd/CMakeLists.txt
@@ -27,6 +27,11 @@ set(LLVM_LINK_COMPONENTS
   FrontendOpenMP
   )
 
+if(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "i386" OR
+   CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "i686")
+  list(APPEND CLANGD_ATOMIC_LIB "atomic")
+endif()
+
 add_clang_library(clangDaemon
   AST.cpp
   ClangdLSPServer.cpp
@@ -114,6 +119,7 @@ add_clang_library(clangDaemon
   clangToolingSyntax
   clangdSupport
   ${LLVM_PTHREAD_LIB}
+  ${CLANGD_ATOMIC_LIB}
   ${ALL_CLANG_TIDY_CHECKS}
   )
 
