Index: llvm-toolchain-13_13.0.0~++20210730102409+6eaf46beb462/clang/tools/libclang/CMakeLists.txt
===================================================================
--- llvm-toolchain-13_13.0.0~++20210730102409+6eaf46beb462.orig/clang/tools/libclang/CMakeLists.txt
+++ llvm-toolchain-13_13.0.0~++20210730102409+6eaf46beb462/clang/tools/libclang/CMakeLists.txt
@@ -48,6 +48,8 @@ set(LIBS
   clangTooling
 )
 
+SET(CLANG_SONAME 13)
+
 if (CLANG_ENABLE_ARCMT)
   list(APPEND LIBS clangARCMigrate)
 endif ()
@@ -171,7 +173,7 @@ if(ENABLE_SHARED)
     # LLVM_VERSION_MAJOR.
     set_target_properties(libclang PROPERTIES
                           VERSION ${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}${LLVM_VERSION_SUFFIX}
-                          SOVERSION 13)
+                          SOVERSION ${CLANG_SONAME})
   endif()
 endif()
 
