Index: llvm-toolchain-9_9.0.0/clang-tools-extra/clang-tidy/tool/run-clang-tidy.py
===================================================================
--- llvm-toolchain-9_9.0.0.orig/clang-tools-extra/clang-tidy/tool/run-clang-tidy.py
+++ llvm-toolchain-9_9.0.0/clang-tools-extra/clang-tidy/tool/run-clang-tidy.py
@@ -180,10 +180,10 @@ def main():
                                    'clang-tidy and clang-apply-replacements in '
                                    '$PATH.')
   parser.add_argument('-clang-tidy-binary', metavar='PATH',
-                      default='clang-tidy',
+                      default='clang-tidy-9',
                       help='path to clang-tidy binary')
   parser.add_argument('-clang-apply-replacements-binary', metavar='PATH',
-                      default='clang-apply-replacements',
+                      default='clang-apply-replacements-9',
                       help='path to clang-apply-replacements binary')
   parser.add_argument('-checks', default=None,
                       help='checks filter, when not specified, use clang-tidy '
