Index: llvm-toolchain-snapshot_12~++20200911085730+f787fe15d8e/clang-tools-extra/clangd/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_12~++20200911085730+f787fe15d8e.orig/clang-tools-extra/clangd/CMakeLists.txt
+++ llvm-toolchain-snapshot_12~++20200911085730+f787fe15d8e/clang-tools-extra/clangd/CMakeLists.txt
@@ -35,6 +35,11 @@ endif()
 
 include_directories(BEFORE "${CMAKE_CURRENT_BINARY_DIR}/../clang-tidy")
 
+if(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "i386" OR
+   CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "i686")
+  list(APPEND CLANGD_ATOMIC_LIB "atomic")
+endif()
+
 add_clang_library(clangDaemon
   AST.cpp
   ClangdLSPServer.cpp
@@ -111,6 +116,7 @@ add_clang_library(clangDaemon
   clangdSupport
   clangTidy
   ${LLVM_PTHREAD_LIB}
+  ${CLANGD_ATOMIC_LIB}
   ${ALL_CLANG_TIDY_CHECKS}
 
   DEPENDS
