Index: llvm-toolchain-snapshot_11~++20200326111000+4673699a470/clang-tools-extra/clangd/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_11~++20200326111000+4673699a470.orig/clang-tools-extra/clangd/CMakeLists.txt
+++ llvm-toolchain-snapshot_11~++20200326111000+4673699a470/clang-tools-extra/clangd/CMakeLists.txt
@@ -34,6 +34,11 @@ if(NOT HAVE_CXX_ATOMICS_WITHOUT_LIB OR N
   list(APPEND CLANGD_ATOMIC_LIB "atomic")
 endif()
 
+if(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "i386" OR
+   CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "i686")
+  list(APPEND CLANGD_ATOMIC_LIB "atomic")
+endif()
+
 add_clang_library(clangDaemon
   AST.cpp
   Cancellation.cpp
