Index: llvm-toolchain-snapshot_16~++20220920091612+a7d2409bac7f/clang-tools-extra/clangd/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_16~++20220920091612+a7d2409bac7f.orig/clang-tools-extra/clangd/CMakeLists.txt
+++ llvm-toolchain-snapshot_16~++20220920091612+a7d2409bac7f/clang-tools-extra/clangd/CMakeLists.txt
@@ -52,6 +52,11 @@ endif()
 
 include_directories(BEFORE "${CMAKE_CURRENT_BINARY_DIR}/../clang-tidy")
 
+if(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "i386" OR
+   CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "i686")
+  list(APPEND CLANGD_ATOMIC_LIB "atomic")
+endif()
+
 add_clang_library(clangDaemon
   AST.cpp
   ASTSignals.cpp
@@ -166,6 +171,7 @@ clang_target_link_libraries(clangDaemon
 target_link_libraries(clangDaemon
   PRIVATE
   ${LLVM_PTHREAD_LIB}
+  ${CLANGD_ATOMIC_LIB}
 
   clangTidy
 
