Index: llvm-toolchain-snapshot_11~++20200714101840+aef60af34ec/clang-tools-extra/clangd/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_11~++20200714101840+aef60af34ec.orig/clang-tools-extra/clangd/CMakeLists.txt
+++ llvm-toolchain-snapshot_11~++20200714101840+aef60af34ec/clang-tools-extra/clangd/CMakeLists.txt
@@ -28,6 +28,11 @@ set(LLVM_LINK_COMPONENTS
   Option
   )
 
+if(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "i386" OR
+   CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "i686")
+  list(APPEND CLANGD_ATOMIC_LIB "atomic")
+endif()
+
 add_clang_library(clangDaemon
   AST.cpp
   ClangdLSPServer.cpp
@@ -104,6 +109,7 @@ add_clang_library(clangDaemon
   clangdSupport
   clangTidy
   ${LLVM_PTHREAD_LIB}
+  ${CLANGD_ATOMIC_LIB}
   ${ALL_CLANG_TIDY_CHECKS}
 
   DEPENDS
