Index: llvm-toolchain-snapshot-10~+201910241319490700d0bd3fc88be/compiler-rt/lib/fuzzer/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-10~+201910241319490700d0bd3fc88be.orig/compiler-rt/lib/fuzzer/CMakeLists.txt
+++ llvm-toolchain-snapshot-10~+201910241319490700d0bd3fc88be/compiler-rt/lib/fuzzer/CMakeLists.txt
@@ -122,7 +122,7 @@ if(OS_NAME MATCHES "Linux|Fuchsia" AND
    COMPILER_RT_LIBCXXABI_PATH)
   macro(partially_link_libcxx name dir arch)
     if(${arch} MATCHES "i386")
-      set(EMULATION_ARGUMENT "-m" "elf_i386")
+      set(EMULATION_ARGUMENT "")
     else()
       set(EMULATION_ARGUMENT "")
     endif()
