Index: llvm-toolchain-build/lib/Support/Unix/Memory.inc
===================================================================
--- llvm-toolchain-build.orig/lib/Support/Unix/Memory.inc	2013-03-15 09:41:15.000000000 +0100
+++ llvm-toolchain-build/lib/Support/Unix/Memory.inc	2013-03-15 09:43:18.000000000 +0100
@@ -329,7 +329,7 @@
   // FIXME: Can we safely always call this for __GNUC__ everywhere?
   const char *Start = static_cast<const char *>(Addr);
   const char *End = Start + Len;
-  __clear_cache(const_cast<char *>(Start), const_cast<char *>(End));
+  __builtin___clear_cache(const_cast<char *>(Start), const_cast<char *>(End));
 #  elif defined(__mips__)
   const char *Start = static_cast<const char *>(Addr);
 #    if defined(ANDROID)
