Index: llvm-toolchain-9-9.0.0/compiler-rt/lib/CMakeLists.txt
===================================================================
--- llvm-toolchain-9-9.0.0.orig/compiler-rt/lib/CMakeLists.txt
+++ llvm-toolchain-9-9.0.0/compiler-rt/lib/CMakeLists.txt
@@ -57,5 +57,8 @@ if(COMPILER_RT_BUILD_XRAY)
 endif()
 
 if(COMPILER_RT_BUILD_LIBFUZZER)
-  compiler_rt_build_runtime(fuzzer)
+  if(NOT CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "i386" AND
+     NOT CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "i686")
+    compiler_rt_build_runtime(fuzzer)
+  endif()
 endif()
Index: llvm-toolchain-9-9.0.0/clang/runtime/CMakeLists.txt
===================================================================
--- llvm-toolchain-9-9.0.0.orig/clang/runtime/CMakeLists.txt
+++ llvm-toolchain-9-9.0.0/clang/runtime/CMakeLists.txt
@@ -130,7 +130,7 @@ if(LLVM_BUILD_EXTERNAL_COMPILER_RT AND E
       FileCheck count not llvm-nm llvm-objdump llvm-symbolizer)
 
     # Add top-level targets for various compiler-rt test suites.
-    set(COMPILER_RT_TEST_SUITES check-fuzzer check-asan check-hwasan check-asan-dynamic check-dfsan
+    set(COMPILER_RT_TEST_SUITES check-asan check-hwasan check-asan-dynamic check-dfsan
       check-lsan check-msan check-sanitizer check-tsan check-ubsan check-ubsan-minimal
       check-profile check-cfi check-cfi-and-supported check-safestack check-gwp_asan)
     foreach(test_suite ${COMPILER_RT_TEST_SUITES})
Index: llvm-toolchain-9-9.0.0/compiler-rt/test/CMakeLists.txt
===================================================================
--- llvm-toolchain-9-9.0.0.orig/compiler-rt/test/CMakeLists.txt
+++ llvm-toolchain-9-9.0.0/compiler-rt/test/CMakeLists.txt
@@ -57,7 +57,10 @@ if(COMPILER_RT_CAN_EXECUTE_TESTS)
 
     # OpenBSD not supporting asan, cannot run the tests
     if(COMPILER_RT_BUILD_LIBFUZZER AND NOT "${CMAKE_SYSTEM_NAME}" STREQUAL "OpenBSD" AND NOT ANDROID)
-      compiler_rt_test_runtime(fuzzer)
+      if(NOT CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "i386" AND
+         NOT CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "i686")
+           compiler_rt_test_runtime(fuzzer)
+      endif()
     endif()
 
     foreach(sanitizer ${COMPILER_RT_SANITIZERS_TO_BUILD})
