Index: llvm-toolchain-snapshot_3.4~svn188990/lldb/lib/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.4~svn188990.orig/lldb/lib/Makefile	2013-08-22 09:13:54.676598924 +0200
+++ llvm-toolchain-snapshot_3.4~svn188990/lldb/lib/Makefile	2013-08-22 09:13:54.672598924 +0200
@@ -154,5 +154,6 @@
     # Allow unresolved symbols.
     LLVMLibsOptions += -Wl,--allow-shlib-undefined
     # Link in python
-    LLVMLibsOptions += $(PYTHON_BUILD_FLAGS) -lrt -L/usr/local/lib -lexecinfo
+    LLVMLibsOptions += $(PYTHON_BUILD_FLAGS) -lrt
+# -L/usr/local/lib -lexecinfo
 endif
