Index: llvm-toolchain-snapshot_10~+20191019070539+bdcad0aca0a/compiler-rt/lib/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_10~+20191019070539+bdcad0aca0a.orig/compiler-rt/lib/CMakeLists.txt
+++ llvm-toolchain-snapshot_10~+20191019070539+bdcad0aca0a/compiler-rt/lib/CMakeLists.txt
@@ -56,6 +56,7 @@ if(COMPILER_RT_BUILD_XRAY)
   compiler_rt_build_runtime(xray)
 endif()
 
 if(COMPILER_RT_BUILD_LIBFUZZER)
-  compiler_rt_build_runtime(fuzzer)
-endif()
+#if(COMPILER_RT_BUILD_LIBFUZZER)
+#  compiler_rt_build_runtime(fuzzer)
+#endif()
Index: llvm-toolchain-snapshot_10~+20191019070539+bdcad0aca0a/compiler-rt/test/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_10~+20191019070539+bdcad0aca0a.orig/compiler-rt/test/CMakeLists.txt
+++ llvm-toolchain-snapshot_10~+20191019070539+bdcad0aca0a/compiler-rt/test/CMakeLists.txt
@@ -53,9 +53,9 @@ if(COMPILER_RT_CAN_EXECUTE_TESTS)
     compiler_rt_test_runtime(sanitizer_common)
 
     # OpenBSD not supporting asan, cannot run the tests
-    if(COMPILER_RT_BUILD_LIBFUZZER AND NOT "${CMAKE_SYSTEM_NAME}" STREQUAL "OpenBSD" AND NOT ANDROID)
-      compiler_rt_test_runtime(fuzzer)
-    endif()
+#    if(COMPILER_RT_BUILD_LIBFUZZER AND NOT "${CMAKE_SYSTEM_NAME}" STREQUAL "OpenBSD" AND NOT ANDROID)
+#      compiler_rt_test_runtime(fuzzer)
+#    endif()
 
     foreach(sanitizer ${COMPILER_RT_SANITIZERS_TO_BUILD})
       # cfi testing is gated on ubsan
