Index: llvm-toolchain-snapshot_3.5~svn199197/lldb/lib/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.5~svn199197.orig/lldb/lib/Makefile	2014-01-14 11:31:05.712868380 +0100
+++ llvm-toolchain-snapshot_3.5~svn199197/lldb/lib/Makefile	2014-01-14 11:31:05.708868380 +0100
@@ -159,5 +159,6 @@
     # Allow unresolved symbols.
     LLVMLibsOptions += -Wl,--allow-shlib-undefined
     # Link in python
-    LLVMLibsOptions += $(PYTHON_BUILD_FLAGS) -lrt -L/usr/local/lib -lexecinfo
+    LLVMLibsOptions += $(PYTHON_BUILD_FLAGS) -lrt
+# -L/usr/local/lib -lexecinfo
 endif
