Index: llvm-toolchain-snapshot_16~++20221111053703+a77a02aa5794/openmp/libompd/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_16~++20221111053703+a77a02aa5794.orig/openmp/libompd/CMakeLists.txt
+++ llvm-toolchain-snapshot_16~++20221111053703+a77a02aa5794/openmp/libompd/CMakeLists.txt
@@ -16,6 +16,6 @@ if(LIBOMP_OMPD_SUPPORT)
     add_subdirectory(src)
     if(LIBOMP_OMPD_GDB_SUPPORT)
         add_subdirectory(gdb-plugin)
-        add_subdirectory(test)
+#        add_subdirectory(test)
     endif()
 endif()
