Index: llvm-toolchain-snapshot_3.5~svn200375/lldb/lib/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.5~svn200375.orig/lldb/lib/Makefile	2014-01-29 07:46:02.284720030 -0800
+++ llvm-toolchain-snapshot_3.5~svn200375/lldb/lib/Makefile	2014-01-29 07:46:02.256720031 -0800
@@ -160,5 +160,6 @@
     # Allow unresolved symbols.
     LLVMLibsOptions += -Wl,--allow-shlib-undefined
     # Link in python
-    LLVMLibsOptions += $(PYTHON_BUILD_FLAGS) -lrt -L/usr/local/lib -lexecinfo
+    LLVMLibsOptions += $(PYTHON_BUILD_FLAGS) -lrt
+# -L/usr/local/lib -lexecinfo
 endif
