Index: llvm-toolchain-9_9.0.0/llvm/utils/lit/lit/Test.py
===================================================================
--- llvm-toolchain-9_9.0.0.orig/llvm/utils/lit/lit/Test.py
+++ llvm-toolchain-9_9.0.0/llvm/utils/lit/lit/Test.py
@@ -32,7 +32,7 @@ PASS        = ResultCode('PASS', False)
 FLAKYPASS   = ResultCode('FLAKYPASS', False)
 XFAIL       = ResultCode('XFAIL', False)
 FAIL        = ResultCode('FAIL', True)
-XPASS       = ResultCode('XPASS', True)
+XPASS       = ResultCode('XPASS', False)
 UNRESOLVED  = ResultCode('UNRESOLVED', True)
 UNSUPPORTED = ResultCode('UNSUPPORTED', False)
 TIMEOUT     = ResultCode('TIMEOUT', True)
