Index: llvm-toolchain-snapshot_3.5~svn195930/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain-snapshot_3.5~svn195930.orig/clang/lib/Driver/ToolChains.cpp	2013-11-28 20:00:36.083311481 +0100
+++ llvm-toolchain-snapshot_3.5~svn195930/clang/lib/Driver/ToolChains.cpp	2013-11-29 11:15:58.145335055 +0100
@@ -134,6 +134,7 @@
     .Default(0);
 }
 
+
 StringRef Darwin::getDarwinArchName(const ArgList &Args) const {
   switch (getTriple().getArch()) {
   default:
@@ -1973,6 +1974,8 @@
       DriverArgs.hasArg(options::OPT_nostdincxx))
     return;
 
+  addSystemInclude(DriverArgs, CC1Args, "/usr/include/clang/" + std::string(CLANG_VERSION_STRING) + "/include/");
+
   switch (GetCXXStdlibType(DriverArgs)) {
   case ToolChain::CST_Libcxx:
     addSystemInclude(DriverArgs, CC1Args,
