Index: llvm-toolchain-snapshot_3.5~svn199197/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain-snapshot_3.5~svn199197.orig/clang/lib/Driver/ToolChains.cpp	2014-01-14 11:31:22.796868045 +0100
+++ llvm-toolchain-snapshot_3.5~svn199197/clang/lib/Driver/ToolChains.cpp	2014-01-14 11:31:22.792868045 +0100
@@ -149,6 +149,7 @@
           A->getValue() == StringRef("soft"));
 }
 
+
 StringRef Darwin::getDarwinArchName(const ArgList &Args) const {
   switch (getTriple().getArch()) {
   default:
@@ -1977,6 +1978,8 @@
       DriverArgs.hasArg(options::OPT_nostdincxx))
     return;
 
+  addSystemInclude(DriverArgs, CC1Args, "/usr/include/clang/" + std::string(CLANG_VERSION_STRING) + "/include/");
+
   switch (GetCXXStdlibType(DriverArgs)) {
   case ToolChain::CST_Libcxx:
     addSystemInclude(DriverArgs, CC1Args,
