Index: llvm-toolchain-snapshot_17~++20230630093049+0446bfcc5ca2/libcxx/src/filesystem/posix_compat.h
===================================================================
--- llvm-toolchain-snapshot_17~++20230630093049+0446bfcc5ca2.orig/libcxx/src/filesystem/posix_compat.h
+++ llvm-toolchain-snapshot_17~++20230630093049+0446bfcc5ca2/libcxx/src/filesystem/posix_compat.h
@@ -39,6 +39,7 @@
 # include <fcntl.h>
 # include <unistd.h>
 # include <sys/stat.h>
+# include <sys/cdefs.h>
 # include <sys/statvfs.h>
 # include <sys/time.h>
 #endif
