Index: llvm-toolchain-snapshot_3.5~svn200375/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain-snapshot_3.5~svn200375.orig/clang/lib/Driver/ToolChains.cpp	2014-01-29 07:46:36.492718958 -0800
+++ llvm-toolchain-snapshot_3.5~svn200375/clang/lib/Driver/ToolChains.cpp	2014-01-29 07:46:36.448718960 -0800
@@ -2107,6 +2107,8 @@
       DriverArgs.hasArg(options::OPT_nostdincxx))
     return;
 
+  addSystemInclude(DriverArgs, CC1Args, "/usr/include/clang/" + std::string(CLANG_VERSION_STRING) + "/include/");
+
   switch (GetCXXStdlibType(DriverArgs)) {
   case ToolChain::CST_Libcxx:
     addSystemInclude(DriverArgs, CC1Args,
