Index: llvm-toolchain-5.0-5.0.2~+rc1/CMakeLists.txt
===================================================================
--- llvm-toolchain-5.0-5.0.2~+rc1.orig/CMakeLists.txt
+++ llvm-toolchain-5.0-5.0.2~+rc1/CMakeLists.txt
@@ -26,7 +26,7 @@ if(NOT DEFINED LLVM_VERSION_MINOR)
   set(LLVM_VERSION_MINOR 0)
 endif()
 if(NOT DEFINED LLVM_VERSION_PATCH)
-  set(LLVM_VERSION_PATCH 1)
+  set(LLVM_VERSION_PATCH 2)
 endif()
 if(NOT DEFINED LLVM_VERSION_SUFFIX)
   set(LLVM_VERSION_SUFFIX "")
