Index: llvm-toolchain-snapshot_3.8~svn247169/tools/llvm-config/llvm-config.cpp
===================================================================
--- llvm-toolchain-snapshot_3.8~svn247169.orig/tools/llvm-config/llvm-config.cpp
+++ llvm-toolchain-snapshot_3.8~svn247169/tools/llvm-config/llvm-config.cpp
@@ -327,9 +327,9 @@ int main(int argc, char **argv) {
       } else if (Arg == "--build-system") {
         OS << LLVM_BUILD_SYSTEM << '\n';
       } else if (Arg == "--obj-root") {
-        OS << ActivePrefix << '\n';
+        OS << ActivePrefix << "/build/" << '\n';
       } else if (Arg == "--src-root") {
-        OS << LLVM_SRC_ROOT << '\n';
+        OS << ActivePrefix << "/build/" << '\n';
       } else {
         usage();
       }
