Index: llvm-toolchain-snapshot_3.8~svn252155/tools/llvm-config/llvm-config.cpp
===================================================================
--- llvm-toolchain-snapshot_3.8~svn252155.orig/tools/llvm-config/llvm-config.cpp
+++ llvm-toolchain-snapshot_3.8~svn252155/tools/llvm-config/llvm-config.cpp
@@ -330,9 +330,9 @@ int main(int argc, char **argv) {
       } else if (Arg == "--has-rtti") {
         OS << LLVM_HAS_RTTI << '\n';
       } else if (Arg == "--obj-root") {
-        OS << ActivePrefix << '\n';
+        OS << ActivePrefix << "/build/" << '\n';
       } else if (Arg == "--src-root") {
-        OS << LLVM_SRC_ROOT << '\n';
+        OS << ActivePrefix << "/build/" << '\n';
       } else {
         usage();
       }
