Index: llvm-toolchain-snapshot_3.8~svn257050/tools/llvm-config/llvm-config.cpp
===================================================================
--- llvm-toolchain-snapshot_3.8~svn257050.orig/tools/llvm-config/llvm-config.cpp
+++ llvm-toolchain-snapshot_3.8~svn257050/tools/llvm-config/llvm-config.cpp
@@ -531,9 +531,9 @@ int main(int argc, char **argv) {
       } else if (Arg == "--shared-mode") {
         PrintSharedMode = true;
       } else if (Arg == "--obj-root") {
-        OS << ActivePrefix << '\n';
+        OS << ActivePrefix << "/build/" << '\n';
       } else if (Arg == "--src-root") {
-        OS << LLVM_SRC_ROOT << '\n';
+        OS << ActivePrefix << "/build/" << '\n';
       } else if (Arg == "--link-shared") {
         LinkMode = LinkModeDyLib;
       } else if (Arg == "--link-static") {
