Index: llvm-toolchain-snapshot-15~++20220703041804+e1f61d864214/compiler-rt/lib/builtins/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-15~++20220703041804+e1f61d864214.orig/compiler-rt/lib/builtins/CMakeLists.txt
+++ llvm-toolchain-snapshot-15~++20220703041804+e1f61d864214/compiler-rt/lib/builtins/CMakeLists.txt
@@ -700,8 +700,11 @@ else ()
   if (COMPILER_RT_HAS_FCF_PROTECTION_FLAG)
     append_list_if(COMPILER_RT_ENABLE_CET -fcf-protection=full BUILTIN_CFLAGS)
   endif()
-
+  if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "x86_64")
+  append_list_if(COMPILER_RT_HAS_FLOAT16 -msse2 -DCOMPILER_RT_HAS_FLOAT16 BUILTIN_CFLAGS)
+  else ()
   append_list_if(COMPILER_RT_HAS_FLOAT16 -DCOMPILER_RT_HAS_FLOAT16 BUILTIN_CFLAGS)
+  endif()
 
   append_list_if(COMPILER_RT_HAS_STD_C11_FLAG -std=c11 BUILTIN_CFLAGS)
 
