Index: llvm-toolchain-snapshot_3.5~svn199197/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain-snapshot_3.5~svn199197.orig/clang/lib/Driver/ToolChains.cpp	2014-01-14 11:31:32.412867856 +0100
+++ llvm-toolchain-snapshot_3.5~svn199197/clang/lib/Driver/ToolChains.cpp	2014-01-14 11:31:32.412867856 +0100
@@ -2685,6 +2685,11 @@
     return;
   }
 
+  // Force the inclusion of the gcc headers (objc/objc.h)
+  addExternCSystemIncludeIfExists(
+     DriverArgs, CC1Args, GCCInstallation.getInstallPath() + "/include");
+
+
   // Lacking those, try to detect the correct set of system includes for the
   // target triple.
 
