Index: llvm-toolchain-snapshot_3.5~svn200375/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain-snapshot_3.5~svn200375.orig/clang/lib/Driver/ToolChains.cpp	2014-01-29 07:48:48.964714807 -0800
+++ llvm-toolchain-snapshot_3.5~svn200375/clang/lib/Driver/ToolChains.cpp	2014-01-29 07:48:48.936714808 -0800
@@ -2839,6 +2839,11 @@
     return;
   }
 
+  // Force the inclusion of the gcc headers (objc/objc.h)
+  addExternCSystemIncludeIfExists(
+     DriverArgs, CC1Args, GCCInstallation.getInstallPath() + "/include");
+
+
   // Lacking those, try to detect the correct set of system includes for the
   // target triple.
 
