Index: llvm-toolchain-snapshot_3.5~svn202473/clang/lib/Basic/FileManager.cpp
===================================================================
--- llvm-toolchain-snapshot_3.5~svn202473.orig/clang/lib/Basic/FileManager.cpp	2014-02-28 09:56:58.459328014 +0100
+++ llvm-toolchain-snapshot_3.5~svn202473/clang/lib/Basic/FileManager.cpp	2014-02-28 09:56:58.455328015 +0100
@@ -476,6 +476,10 @@
   UniqueRealFiles.erase(Entry->getUniqueID());
 }
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
 
 void FileManager::GetUniqueIDMapping(
                    SmallVectorImpl<const FileEntry *> &UIDToFiles) const {
Index: llvm-toolchain-snapshot_3.5~svn202473/lldb/include/lldb/lldb-defines.h
===================================================================
--- llvm-toolchain-snapshot_3.5~svn202473.orig/lldb/include/lldb/lldb-defines.h	2014-02-28 09:56:58.459328014 +0100
+++ llvm-toolchain-snapshot_3.5~svn202473/lldb/include/lldb/lldb-defines.h	2014-02-28 09:56:58.455328015 +0100
@@ -12,6 +12,11 @@
 
 #include "lldb/lldb-types.h"
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 #if !defined(UINT32_MAX)
     #define UINT32_MAX 4294967295U
 #endif
Index: llvm-toolchain-snapshot_3.5~svn202473/lib/Support/Unix/Path.inc
===================================================================
--- llvm-toolchain-snapshot_3.5~svn202473.orig/lib/Support/Unix/Path.inc	2014-02-28 09:56:58.459328014 +0100
+++ llvm-toolchain-snapshot_3.5~svn202473/lib/Support/Unix/Path.inc	2014-02-28 09:56:58.455328015 +0100
@@ -58,6 +58,7 @@
 // For GNU Hurd
 #if defined(__GNU__) && !defined(PATH_MAX)
 # define PATH_MAX 4096
+# define MAXPATHLEN 4096
 #endif
 
 using namespace llvm;
