Index: llvm-toolchain-snapshot_4.0~svn278655/clang/lib/Basic/FileManager.cpp
===================================================================
--- llvm-toolchain-snapshot_4.0~svn278655.orig/clang/lib/Basic/FileManager.cpp
+++ llvm-toolchain-snapshot_4.0~svn278655/clang/lib/Basic/FileManager.cpp
@@ -500,6 +500,12 @@ void FileManager::invalidateCache(const
   UniqueRealFiles.erase(Entry->getUniqueID());
 }
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
+
 void FileManager::GetUniqueIDMapping(
                    SmallVectorImpl<const FileEntry *> &UIDToFiles) const {
   UIDToFiles.clear();
Index: llvm-toolchain-snapshot_4.0~svn278655/lldb/include/lldb/lldb-defines.h
===================================================================
--- llvm-toolchain-snapshot_4.0~svn278655.orig/lldb/include/lldb/lldb-defines.h
+++ llvm-toolchain-snapshot_4.0~svn278655/lldb/include/lldb/lldb-defines.h
@@ -28,6 +28,11 @@
     #define INT32_MAX 2147483647
 #endif
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 #if !defined(UINT32_MAX)
     #define UINT32_MAX 4294967295U
 #endif
Index: llvm-toolchain-snapshot_4.0~svn278655/lib/Support/Unix/Path.inc
===================================================================
--- llvm-toolchain-snapshot_4.0~svn278655.orig/lib/Support/Unix/Path.inc
+++ llvm-toolchain-snapshot_4.0~svn278655/lib/Support/Unix/Path.inc
@@ -62,6 +62,7 @@
 // For GNU Hurd
 #if defined(__GNU__) && !defined(PATH_MAX)
 # define PATH_MAX 4096
+# define MAXPATHLEN 4096
 #endif
 
 #include <sys/types.h>
Index: llvm-toolchain-snapshot_4.0~svn278655/tools/dsymutil/DwarfLinker.cpp
===================================================================
--- llvm-toolchain-snapshot_4.0~svn278655.orig/tools/dsymutil/DwarfLinker.cpp
+++ llvm-toolchain-snapshot_4.0~svn278655/tools/dsymutil/DwarfLinker.cpp
@@ -41,6 +41,11 @@
 #include <string>
 #include <tuple>
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 namespace llvm {
 namespace dsymutil {
 
