Index: llvm-toolchain-snapshot_8~svn342269/clang/lib/Basic/FileManager.cpp
===================================================================
--- llvm-toolchain-snapshot_8~svn342269.orig/clang/lib/Basic/FileManager.cpp
+++ llvm-toolchain-snapshot_8~svn342269/clang/lib/Basic/FileManager.cpp
@@ -511,6 +511,12 @@ void FileManager::invalidateCache(const
   UniqueRealFiles.erase(Entry->getUniqueID());
 }
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
+
 void FileManager::GetUniqueIDMapping(
                    SmallVectorImpl<const FileEntry *> &UIDToFiles) const {
   UIDToFiles.clear();
Index: llvm-toolchain-snapshot_8~svn342269/lldb/include/lldb/lldb-defines.h
===================================================================
--- llvm-toolchain-snapshot_8~svn342269.orig/lldb/include/lldb/lldb-defines.h
+++ llvm-toolchain-snapshot_8~svn342269/lldb/include/lldb/lldb-defines.h
@@ -28,6 +28,11 @@
 #define INT32_MAX 2147483647
 #endif
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 #if !defined(UINT32_MAX)
 #define UINT32_MAX 4294967295U
 #endif
Index: llvm-toolchain-snapshot_8~svn342269/lib/Support/Unix/Path.inc
===================================================================
--- llvm-toolchain-snapshot_8~svn342269.orig/lib/Support/Unix/Path.inc
+++ llvm-toolchain-snapshot_8~svn342269/lib/Support/Unix/Path.inc
@@ -49,6 +49,7 @@
 // For GNU Hurd
 #if defined(__GNU__) && !defined(PATH_MAX)
 # define PATH_MAX 4096
+# define MAXPATHLEN 4096
 #endif
 
 #include <sys/types.h>
Index: llvm-toolchain-snapshot_8~svn342269/tools/dsymutil/DwarfLinker.cpp
===================================================================
--- llvm-toolchain-snapshot_8~svn342269.orig/tools/dsymutil/DwarfLinker.cpp
+++ llvm-toolchain-snapshot_8~svn342269/tools/dsymutil/DwarfLinker.cpp
@@ -100,6 +100,11 @@
 #include <utility>
 #include <vector>
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 namespace llvm {
 namespace dsymutil {
 
Index: llvm-toolchain-snapshot_8~svn342269/polly/lib/External/ppcg/cuda_common.c
===================================================================
--- llvm-toolchain-snapshot_8~svn342269.orig/polly/lib/External/ppcg/cuda_common.c
+++ llvm-toolchain-snapshot_8~svn342269/polly/lib/External/ppcg/cuda_common.c
@@ -15,6 +15,11 @@
 #include "cuda_common.h"
 #include "ppcg.h"
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 /* Open the host .cu file and the kernel .hu and .cu files for writing.
  * Add the necessary includes.
  */
