Index: llvm-toolchain-snapshot_3.4~svn184290/lib/ExecutionEngine/JIT/JIT.cpp
===================================================================
--- llvm-toolchain-snapshot_3.4~svn184290.orig/lib/ExecutionEngine/JIT/JIT.cpp	2013-06-19 11:14:03.050779689 +0200
+++ llvm-toolchain-snapshot_3.4~svn184290/lib/ExecutionEngine/JIT/JIT.cpp	2013-06-19 11:14:03.046779699 +0200
@@ -68,7 +68,7 @@
 }
 
 // Determine whether we can register EH tables.
-#if (defined(__GNUC__) && !defined(__ARM_EABI__) && \
+#if (defined(__GNUC__) && !defined(__ARM_EABI__) && !defined(__ia64__) && \
      !defined(__USING_SJLJ_EXCEPTIONS__))
 #define HAVE_EHTABLE_SUPPORT 1
 #else
