Index: llvm-toolchain-snapshot_9~svn351647/lib/Support/Unix/Path.inc
===================================================================
--- llvm-toolchain-snapshot_9~svn351647.orig/lib/Support/Unix/Path.inc
+++ llvm-toolchain-snapshot_9~svn351647/lib/Support/Unix/Path.inc
@@ -176,7 +176,7 @@ std::string getMainExecutable(const char
 
   if (getprogpath(exe_path, argv0) != NULL)
     return exe_path;
-#elif defined(__linux__) || defined(__CYGWIN__)
+#elif defined(__linux__) || defined(__CYGWIN__) || defined(__GNU__)
   char exe_path[MAXPATHLEN];
   StringRef aPath("/proc/self/exe");
   if (sys::fs::exists(aPath)) {
