Index: llvm-toolchain_3.3~svn179680/lldb/source/Plugins/DynamicLoader/MacOSX-DYLD/DynamicLoaderMacOSXDYLD.cpp
===================================================================
--- llvm-toolchain_3.3~svn179680.orig/lldb/source/Plugins/DynamicLoader/MacOSX-DYLD/DynamicLoaderMacOSXDYLD.cpp	2013-04-16 08:28:00.000000000 +0200
+++ llvm-toolchain_3.3~svn179680/lldb/source/Plugins/DynamicLoader/MacOSX-DYLD/DynamicLoaderMacOSXDYLD.cpp	2013-04-17 12:17:32.311258764 +0200
@@ -30,6 +30,8 @@
 
 #include "DynamicLoaderMacOSXDYLD.h"
 
+#include <uuid/uuid.h>
+
 //#define ENABLE_DEBUG_PRINTF // COMMENT THIS LINE OUT PRIOR TO CHECKIN
 #ifdef ENABLE_DEBUG_PRINTF
 #include <stdio.h>
