Index: llvm-toolchain-snapshot_3.4~svn193982/lldb/source/Core/Mangled.cpp
===================================================================
--- llvm-toolchain-snapshot_3.4~svn193982.orig/lldb/source/Core/Mangled.cpp	2013-11-04 09:46:37.000668903 +0100
+++ llvm-toolchain-snapshot_3.4~svn193982/lldb/source/Core/Mangled.cpp	2013-11-04 09:46:36.996668902 +0100
@@ -4712,6 +4712,7 @@
 #include <ctype.h>
 #include <string.h>
 #include <stdlib.h>
+#include <cstdlib>
 
 using namespace lldb_private;
 
