Index: llvm-toolchain-snapshot_3.4~svn193603/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain-snapshot_3.4~svn193603.orig/clang/lib/Driver/ToolChains.cpp	2013-10-29 12:26:35.270138047 +0100
+++ llvm-toolchain-snapshot_3.4~svn193603/clang/lib/Driver/ToolChains.cpp	2013-10-29 12:27:25.930139892 +0100
@@ -2707,6 +2707,12 @@
   StringRef BiarchSuffix = GCCInstallation.getBiarchSuffix();
   const GCCVersion &Version = GCCInstallation.getVersion();
 
+#ifdef __i386__
+  // clang does not detect this path: /usr/include/i386-linux-gnu/c++/4.8/ Helphim
+// For now, comment this patch to make sure it is fixed upstream
+//  addLibStdCXXIncludePaths( "/usr/include/", "/c++/" + Version.str(), "i386-linux-gnu", GCCInstallation.getMultiarchSuffix(), DriverArgs, CC1Args);
+#endif
+
   if (addLibStdCXXIncludePaths(LibDir.str() + "/../include",
                                "/c++/" + Version.Text, TripleStr, BiarchSuffix,
                                MIPSABIDirSuffix, DriverArgs, CC1Args))
