Index: llvm-toolchain-3.9-3.9.1/lldb/source/Plugins/Process/gdb-remote/GDBRemoteCommunication.cpp
===================================================================
--- llvm-toolchain-3.9-3.9.1.orig/lldb/source/Plugins/Process/gdb-remote/GDBRemoteCommunication.cpp
+++ llvm-toolchain-3.9-3.9.1/lldb/source/Plugins/Process/gdb-remote/GDBRemoteCommunication.cpp
@@ -40,7 +40,7 @@
 #if defined(__APPLE__)
 # define DEBUGSERVER_BASENAME    "debugserver"
 #else
-# define DEBUGSERVER_BASENAME    "lldb-server"
+# define DEBUGSERVER_BASENAME    "lldb-server-3.9"
 #endif
 
 #if defined (HAVE_LIBCOMPRESSION)
@@ -1175,7 +1175,7 @@ GDBRemoteCommunication::StartDebugserver
             }
         }
     }
-    
+    log->Printf("foo %s,  debugserver_file_spec.GetPath ().c_str ());
     if (debugserver_exists)
     {
         debugserver_file_spec.GetPath (debugserver_path, sizeof(debugserver_path));
