Index: llvm-toolchain-snapshot_4.0~svn280802/lldb/scripts/Python/finishSwigPythonLLDB.py
===================================================================
--- llvm-toolchain-snapshot_4.0~svn280802.orig/lldb/scripts/Python/finishSwigPythonLLDB.py
+++ llvm-toolchain-snapshot_4.0~svn280802/lldb/scripts/Python/finishSwigPythonLLDB.py
@@ -443,7 +443,7 @@ def make_symlink_liblldb(
             if eOSType == utilsOsType.EnumOsType.Darwin:
                 strLibFileExtn = ".dylib"
             else:
-                strLibFileExtn = ".so"
+                strLibFileExtn = "-4.0.so"
             strSrc = os.path.join(vstrLldbLibDir, "liblldb" + strLibFileExtn)
 
     bOk, strErrMsg = make_symlink(
