---
 lldb/source/API/CMakeLists.txt |   14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

Index: llvm-toolchain-snapshot_7~svn338163/lldb/source/API/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_7~svn338163.orig/lldb/source/API/CMakeLists.txt
+++ llvm-toolchain-snapshot_7~svn338163/lldb/source/API/CMakeLists.txt
@@ -4,7 +4,12 @@ endif()
 
 get_property(LLDB_ALL_PLUGINS GLOBAL PROPERTY LLDB_PLUGINS)
 
-add_lldb_library(liblldb SHARED
+set(output_name lldb)
+if (CMAKE_SYSTEM_NAME MATCHES "Windows")
+  set(output_name liblldb)
+endif()
+
+add_lldb_library(liblldb SHARED OUTPUT_NAME ${output_name} SONAME
   SBAddress.cpp
   SBAttachInfo.cpp
   SBBlock.cpp
@@ -121,7 +126,7 @@ if (LLDB_BUILD_FRAMEWORK)
 else()
   set_target_properties(liblldb
     PROPERTIES
-    VERSION ${LLDB_VERSION}
+    VERSION 1
     OUTPUT_NAME lldb)
 endif()
 
