---
 lldb/source/API/CMakeLists.txt |   14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

Index: llvm-toolchain-snapshot_8~svn350421/lldb/source/API/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_8~svn350421.orig/lldb/source/API/CMakeLists.txt
+++ llvm-toolchain-snapshot_8~svn350421/lldb/source/API/CMakeLists.txt
@@ -9,7 +9,12 @@ if(NOT LLDB_DISABLE_PYTHON)
   set(lldb_python_wrapper ${lldb_scripts_dir}/LLDBWrapPython.cpp)
 endif()
 
-add_lldb_library(liblldb SHARED
+set(output_name lldb)
+if (CMAKE_SYSTEM_NAME MATCHES "Windows")
+  set(output_name liblldb)
+endif()
+
+add_lldb_library(liblldb SHARED OUTPUT_NAME ${output_name} SONAME
   SBAddress.cpp
   SBAttachInfo.cpp
   SBBlock.cpp
@@ -120,7 +125,7 @@ endif()
 
 set_target_properties(liblldb
   PROPERTIES
-  VERSION ${LLDB_VERSION}
+  VERSION 1
 )
 
 if (NOT CMAKE_SYSTEM_NAME MATCHES "Windows")
@@ -144,11 +149,6 @@ if ( CMAKE_SYSTEM_NAME MATCHES "Windows"
   if (MSVC AND NOT LLDB_DISABLE_PYTHON)
     target_link_libraries(liblldb PRIVATE ${PYTHON_LIBRARY})
   endif()
-else()
-  set_target_properties(liblldb
-    PROPERTIES
-    OUTPUT_NAME lldb
-  )
 endif()
 
 if(LLDB_BUILD_FRAMEWORK)
