diff --git llvm-toolchain-snapshot_3.5~svn201241.orig/lldb/tools/lldb-mi/MICmnLogMediumFile.cpp llvm-toolchain-snapshot_3.5~svn201241/lldb/tools/lldb-mi/MICmnLogMediumFile.cpp
index a14ad04..382db07 100644
--- llvm-toolchain-snapshot_3.5~svn201241.orig/tools/lldb-mi/MICmnLogMediumFile.cpp
+++ llvm-toolchain-snapshot_3.5~svn201241/lldb/tools/lldb-mi/MICmnLogMediumFile.cpp
@@ -28,7 +28,7 @@
 
 #if defined( _MSC_VER )
 	#include "MIUtilSystemWindows.h"
-#elif defined( __FreeBSD__ ) || defined( __linux )
+#elif defined( __FreeBSD__ ) || defined( __linux__ )
 	#include "MIUtilSystemLinux.h"
 #elif defined( __APPLE__ )
 	#include "MIUtilSystemOsx.h"
diff --git llvm-toolchain-snapshot_3.5~svn201241.orig/lldb//tools/lldb-mi/MIUtilSystemLinux.cpp llvm-toolchain-snapshot_3.5~svn201241/lldb//tools/lldb-mi/MIUtilSystemLinux.cpp
index de14317..9c4deb4 100644
--- llvm-toolchain-snapshot_3.5~svn201241.orig/lldb/tools/lldb-mi/MIUtilSystemLinux.cpp
+++ llvm-toolchain-snapshot_3.5~svn201241/lldb/tools/lldb-mi/MIUtilSystemLinux.cpp
@@ -22,7 +22,7 @@
 // Include compiler configuration
 #include "MICmnConfig.h"
 
-#if defined( __FreeBSD__ ) || defined( __linux )
+#if defined( __FreeBSD__ ) || defined( __linux__ )
 
 // In-house headers:
 #include "MIUtilSystemLinux.h"
@@ -119,4 +119,4 @@ bool CMIUtilSystemLinux::GetLogFilesPath( CMIUtilString & vrwFileNamePath ) cons
 	return MIstatus::success;
 }
 
-#endif // #if defined( __linux )
+#endif // #if defined( __linux__ )
diff --git llvm-toolchain-snapshot_3.5~svn201241/lldb/tools/lldb-mi/MIUtilSystemLinux.h llvm-toolchain-snapshot_3.5~svn201241/lldb//tools/lldb-mi/MIUtilSystemLinux.h
index c79bcfd..0d9062a 100644
--- llvm-toolchain-snapshot_3.5~svn201241.orig/lldb/tools/lldb-mi/MIUtilSystemLinux.h
+++ llvm-toolchain-snapshot_3.5~svn201241/lldb/tools/lldb-mi/MIUtilSystemLinux.h
@@ -24,7 +24,7 @@
 // Include compiler configuration
 #include "MICmnConfig.h"
 
-#if defined( __FreeBSD__ ) || defined( __linux )
+#if defined( __FreeBSD__ ) || defined( __linux__ )
 
 // In-house headers:
 #include "MIUtilString.h"  
@@ -57,4 +57,4 @@ public:
 
 typedef CMIUtilSystemLinux CMIUtilSystem;
 
-#endif // #if defined( __linux )
+#endif // #if defined( __linux__ )
-- 
2.0.1

