Index: llvm-toolchain-snapshot_3.8~svn243867/lldb/source/Plugins/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.8~svn243867.orig/lldb/source/Plugins/Makefile
+++ llvm-toolchain-snapshot_3.8~svn243867/lldb/source/Plugins/Makefile
@@ -42,12 +42,14 @@ PARALLEL_DIRS += SymbolVendor/MacOSX
 PARALLEL_DIRS += Process/mach-core
 PARALLEL_DIRS += Process/elf-core
 PARALLEL_DIRS += JITLoader/GDB
+PARALLEL_DIRS += ScriptInterpreter/Python ScriptInterpreter/None
 endif
 
 ifeq ($(HOST_OS),Linux)
 PARALLEL_DIRS += Process/Linux Process/POSIX
 PARALLEL_DIRS += Process/elf-core
 PARALLEL_DIRS += JITLoader/GDB
+PARALLEL_DIRS += ScriptInterpreter/Python ScriptInterpreter/None
 endif
 
 ifeq ($(HOST_OS),MingW)
Index: llvm-toolchain-snapshot_3.8~svn243867/lldb/source/Plugins/ScriptInterpreter/None/Makefile
===================================================================
--- /dev/null
+++ llvm-toolchain-snapshot_3.8~svn243867/lldb/source/Plugins/ScriptInterpreter/None/Makefile
@@ -0,0 +1,14 @@
+##===- source/Plugins/ScriptInterpreter/None/Makefile ------------*- Makefile -*-===##
+# 
+#                     The LLVM Compiler Infrastructure
+#
+# This file is distributed under the University of Illinois Open Source
+# License. See LICENSE.TXT for details.
+# 
+##===----------------------------------------------------------------------===##
+
+LLDB_LEVEL := ../../../..
+LIBRARYNAME := lldbPluginScriptInterpreterNone
+BUILD_ARCHIVE = 1
+
+include $(LLDB_LEVEL)/Makefile
Index: llvm-toolchain-snapshot_3.8~svn243867/lldb/source/Plugins/ScriptInterpreter/Python/Makefile
===================================================================
--- /dev/null
+++ llvm-toolchain-snapshot_3.8~svn243867/lldb/source/Plugins/ScriptInterpreter/Python/Makefile
@@ -0,0 +1,14 @@
+##===- source/Plugins/ScriptInterpreter/Python/Makefile ------------*- Makefile -*-===##
+# 
+#                     The LLVM Compiler Infrastructure
+#
+# This file is distributed under the University of Illinois Open Source
+# License. See LICENSE.TXT for details.
+# 
+##===----------------------------------------------------------------------===##
+
+LLDB_LEVEL := ../../../..
+LIBRARYNAME := lldbPluginScriptInterpreterPython
+BUILD_ARCHIVE = 1
+
+include $(LLDB_LEVEL)/Makefile
Index: llvm-toolchain-snapshot_3.8~svn243867/lldb/lib/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.8~svn243867.orig/lldb/lib/Makefile
+++ llvm-toolchain-snapshot_3.8~svn243867/lldb/lib/Makefile
@@ -120,14 +120,18 @@ ifeq ($(HOST_OS),Darwin)
               lldbPluginProcessDarwin.a \
               lldbPluginProcessMachCore.a \
               lldbPluginProcessElfCore.a \
-              lldbPluginJITLoaderGDB.a
+              lldbPluginJITLoaderGDB.a \
+              lldbPluginScriptInterpreterNone.a \
+              lldbPluginScriptInterpreterPython.a
 endif
 
 ifeq ($(HOST_OS),Linux)
   USEDLIBS += lldbPluginProcessLinux.a \
               lldbPluginProcessPOSIX.a \
               lldbPluginProcessElfCore.a \
-              lldbPluginJITLoaderGDB.a
+              lldbPluginJITLoaderGDB.a \
+              lldbPluginScriptInterpreterNone.a \
+              lldbPluginScriptInterpreterPython.a
 endif
 
 ifeq ($(HOST_OS),MingW)
