Index: llvm-toolchain-3.9-3.9.1/lldb/source/Plugins/Process/gdb-remote/GDBRemoteCommunication.cpp
===================================================================
--- llvm-toolchain-3.9-3.9.1.orig/lldb/source/Plugins/Process/gdb-remote/GDBRemoteCommunication.cpp
+++ llvm-toolchain-3.9-3.9.1/lldb/source/Plugins/Process/gdb-remote/GDBRemoteCommunication.cpp
@@ -37,14 +37,10 @@
 // Project includes
 #include "ProcessGDBRemoteLog.h"
 
-#ifndef DEBIAN_VERSION_SUFFIX
-#define DEBIAN_VERSION_SUFFIX ""
-#endif
-
 #if defined(__APPLE__)
 # define DEBUGSERVER_BASENAME    "debugserver"
 #else
-# define DEBUGSERVER_BASENAME    "lldb-server" DEBIAN_VERSION_SUFFIX
+# define DEBUGSERVER_BASENAME    "lldb-server-3.9"
 #endif
 
 #if defined (HAVE_LIBCOMPRESSION)
