---
 lldb/source/API/CMakeLists.txt |   14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

Index: llvm-toolchain-snapshot_9~svn365742/lldb/source/API/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_9~svn365742.orig/lldb/source/API/CMakeLists.txt
+++ llvm-toolchain-snapshot_9~svn365742/lldb/source/API/CMakeLists.txt
@@ -13,7 +13,12 @@ if(LLDB_BUILD_FRAMEWORK)
   set(option_install_prefix INSTALL_PREFIX ${LLDB_FRAMEWORK_INSTALL_DIR})
 endif()
 
-add_lldb_library(liblldb SHARED
+set(output_name lldb)
+if (CMAKE_SYSTEM_NAME MATCHES "Windows")
+  set(output_name liblldb)
+endif()
+
+add_lldb_library(liblldb SHARED OUTPUT_NAME ${output_name} SONAME
   SBAddress.cpp
   SBAttachInfo.cpp
   SBBlock.cpp
@@ -130,7 +135,7 @@ endif()
 
 set_target_properties(liblldb
   PROPERTIES
-  VERSION ${LLDB_VERSION}
+  VERSION 1
 )
 
 if (NOT CMAKE_SYSTEM_NAME MATCHES "Windows")
@@ -155,11 +160,6 @@ if ( CMAKE_SYSTEM_NAME MATCHES "Windows"
   if (MSVC AND NOT LLDB_DISABLE_PYTHON)
     target_link_libraries(liblldb PRIVATE ${PYTHON_LIBRARY})
   endif()
-else()
-  set_target_properties(liblldb
-    PROPERTIES
-    OUTPUT_NAME lldb
-  )
 endif()
 
 if (NOT LLDB_BUILT_STANDALONE)
