---
 lldb/source/API/CMakeLists.txt |   14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

Index: llvm-toolchain-snapshot_10~+20191113090841+5b9e4daef06/lldb/source/API/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_10~+20191113090841+5b9e4daef06.orig/lldb/source/API/CMakeLists.txt
+++ llvm-toolchain-snapshot_10~+20191113090841+5b9e4daef06/lldb/source/API/CMakeLists.txt
@@ -14,7 +14,12 @@ if(LLDB_BUILD_FRAMEWORK)
   set(option_framework FRAMEWORK)
 endif()
 
-add_lldb_library(liblldb SHARED ${option_framework}
+set(output_name lldb)
+if (CMAKE_SYSTEM_NAME MATCHES "Windows")
+  set(output_name liblldb)
+endif()
+
+add_lldb_library(liblldb SHARED OUTPUT_NAME ${output_name} SONAME
   SBAddress.cpp
   SBAttachInfo.cpp
   SBBlock.cpp
@@ -132,7 +137,7 @@ endif()
 
 set_target_properties(liblldb
   PROPERTIES
-  VERSION ${LLDB_VERSION}
+  VERSION 1
 )
 
 if (NOT CMAKE_SYSTEM_NAME MATCHES "Windows")
@@ -157,11 +162,6 @@ if ( CMAKE_SYSTEM_NAME MATCHES "Windows"
   if (MSVC AND NOT LLDB_DISABLE_PYTHON)
     target_link_libraries(liblldb PRIVATE ${PYTHON_LIBRARY})
   endif()
-else()
-  set_target_properties(liblldb
-    PROPERTIES
-    OUTPUT_NAME lldb
-  )
 endif()
 
 if (NOT LLDB_BUILT_STANDALONE)
