Index: llvm-toolchain-snapshot_3.7~svn241962/test/lit.cfg
===================================================================
--- llvm-toolchain-snapshot_3.7~svn241962.orig/test/lit.cfg
+++ llvm-toolchain-snapshot_3.7~svn241962/test/lit.cfg
@@ -366,7 +366,7 @@ def have_ld_plugin_support():
     if 'elf32ppc' in emulations:
         config.available_features.add('ld_emu_elf32ppc')
 
-    ld_version = subprocess.Popen([config.gold_executable, '--version'], stdout = subprocess.PIPE)
+    ld_version = subprocess.Popen([config.gold_executable, '--version'], stdout = subprocess.PIPE, env={'LANG': 'C'})
     if not 'GNU gold' in ld_version.stdout.read().decode():
         return False
     ld_version.wait()
