Index: llvm-toolchain-snapshot_13~++20210516062812+d539357e1b8a/llvm/cmake/modules/HandleLLVMOptions.cmake
===================================================================
--- llvm-toolchain-snapshot_13~++20210516062812+d539357e1b8a.orig/llvm/cmake/modules/HandleLLVMOptions.cmake
+++ llvm-toolchain-snapshot_13~++20210516062812+d539357e1b8a/llvm/cmake/modules/HandleLLVMOptions.cmake
@@ -1194,8 +1194,8 @@ if(LLVM_USE_RELATIVE_PATHS_IN_FILES)
 endif()
 
 if(LLVM_INCLUDE_TESTS)
-  # Lit test suite requires at least python 3.6
-  set(LLVM_MINIMUM_PYTHON_VERSION 3.6)
+  # Lit test suite requires at least python 3.5
+  set(LLVM_MINIMUM_PYTHON_VERSION 3.5)
 else()
   # FIXME: it is unknown if this is the actual minimum bound
   set(LLVM_MINIMUM_PYTHON_VERSION 3.0)
