Index: llvm-toolchain-snapshot_15~++20220314071859+3b61587c9e27/llvm/cmake/modules/HandleLLVMOptions.cmake
===================================================================
--- llvm-toolchain-snapshot_15~++20220314071859+3b61587c9e27.orig/llvm/cmake/modules/HandleLLVMOptions.cmake
+++ llvm-toolchain-snapshot_15~++20220314071859+3b61587c9e27/llvm/cmake/modules/HandleLLVMOptions.cmake
@@ -1236,8 +1236,8 @@ if(LLVM_USE_RELATIVE_PATHS_IN_FILES)
 endif()
 
 if(LLVM_INCLUDE_TESTS)
-  # Lit test suite requires at least python 3.6
-  set(LLVM_MINIMUM_PYTHON_VERSION 3.6)
+  # Lit test suite requires at least python 3.5
+  set(LLVM_MINIMUM_PYTHON_VERSION 3.5)
 else()
   # FIXME: it is unknown if this is the actual minimum bound
   set(LLVM_MINIMUM_PYTHON_VERSION 3.0)
