---
 clang/lib/Driver/Tools.cpp |    5 -----
 1 file changed, 5 deletions(-)

Index: llvm-toolchain-snapshot_5.0~svn296106/clang/lib/Driver/Arch/Mips.cpp
===================================================================
--- llvm-toolchain-snapshot_5.0~svn296106.orig/clang/lib/Driver/Arch/Mips.cpp
+++ llvm-toolchain-snapshot_5.0~svn296106/clang/lib/Driver/Arch/Mips.cpp
@@ -362,10 +362,6 @@ bool mips::isFP64ADefault(const llvm::Tr
 
 bool mips::isFPXXDefault(const llvm::Triple &Triple, StringRef CPUName,
                          StringRef ABIName, mips::FloatABI FloatABI) {
-  if (Triple.getVendor() != llvm::Triple::ImaginationTechnologies &&
-      Triple.getVendor() != llvm::Triple::MipsTechnologies &&
-      !Triple.isAndroid())
-    return false;
 
   if (ABIName != "32")
     return false;
