--- llvm-toolchain-16-16.0.6.orig/openmp/runtime/cmake/config-ix.cmake
+++ llvm-toolchain-16-16.0.6/openmp/runtime/cmake/config-ix.cmake
@@ -328,6 +329,7 @@ else()
       (LIBOMP_ARCH STREQUAL aarch64) OR
       (LIBOMP_ARCH STREQUAL aarch64_a64fx) OR
       (LIBOMP_ARCH STREQUAL ppc64le) OR
+      (LIBOMP_ARCH STREQUAL mips64) OR
       (LIBOMP_ARCH STREQUAL ppc64) OR
       (LIBOMP_ARCH STREQUAL riscv64) OR
       (LIBOMP_ARCH STREQUAL loongarch64))
