Index: llvm-toolchain-snapshot_4.0~svn278527/clang/lib/Driver/Tools.cpp
===================================================================
--- llvm-toolchain-snapshot_4.0~svn278527.orig/clang/lib/Driver/Tools.cpp
+++ llvm-toolchain-snapshot_4.0~svn278527/clang/lib/Driver/Tools.cpp
@@ -7391,11 +7391,6 @@ bool mips::isFP64ADefault(const llvm::Tr
 
 bool mips::isFPXXDefault(const llvm::Triple &Triple, StringRef CPUName,
                          StringRef ABIName, mips::FloatABI FloatABI) {
-  if (Triple.getVendor() != llvm::Triple::ImaginationTechnologies &&
-      Triple.getVendor() != llvm::Triple::MipsTechnologies &&
-      !Triple.isAndroid())
-    return false;
-
   if (ABIName != "32")
     return false;
 
