Index: llvm-toolchain-snapshot_3.5~svn195268/bindings/ocaml/backends/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.5~svn195268.orig/bindings/ocaml/backends/Makefile	(révision 195335)
+++ llvm-toolchain-snapshot_3.5~svn195268/bindings/ocaml/backends/Makefile	(révision 195336)
@@ -53,4 +53,9 @@
 		$(MAKE) -f Makefile.$$i uninstall; \
 	done
 
-.PHONY: all-backends clean-backends install-backends uninstall-backends
+ocamldoc: stubs
+	$(Verb) for i in $(TARGETS_TO_BUILD); do \
+		$(MAKE) -f Makefile.$$i ocamldoc; \
+	done
+
+.PHONY: all-backends clean-backends install-backends uninstall-backends ocamldoc
