Index: llvm-toolchain-snapshot_8~svn339515/openmp/libomptarget/src/omptarget.cpp
===================================================================
--- llvm-toolchain-snapshot_8~svn339515.orig/openmp/libomptarget/src/omptarget.cpp
+++ llvm-toolchain-snapshot_8~svn339515/openmp/libomptarget/src/omptarget.cpp
@@ -13,6 +13,8 @@
 //===----------------------------------------------------------------------===//
 
 #include <omptarget.h>
+// /build/llvm-toolchain-snapshot-8~svn340542/projects/openmp/libomptarget/src/omptarget.cpp:37:21: error: 'va_start' was not declared in this scope
+#include <cstdarg>
 
 #include "device.h"
 #include "private.h"
