Index: llvm-toolchain-build/polly/lib/CodeGen/LoopGenerators.cpp
===================================================================
--- llvm-toolchain-build.orig/polly/lib/CodeGen/LoopGenerators.cpp	2013-03-15 09:40:09.000000000 +0100
+++ llvm-toolchain-build/polly/lib/CodeGen/LoopGenerators.cpp	2013-03-15 09:43:18.000000000 +0100
@@ -203,7 +203,7 @@
   for (unsigned i = 0; i < OldValues.size(); i++) {
     Value *Address = Builder.CreateStructGEP(Struct, i);
     Value *NewValue = Builder.CreateLoad(Address);
-    Map.insert(std::make_pair<Value *, Value *>(OldValues[i], NewValue));
+    Map.insert(std::pair<Value*, Value*>(OldValues[i], NewValue));
   }
 }
 
