Author: Gianfranco Costamagna <locutusofborg@debian.org>
Last-Update: 2020-01-21

Index: llvm-toolchain-snapshot_15~++20220314071859+3b61587c9e27/lldb/source/Plugins/ScriptInterpreter/Python/ScriptInterpreterPython.cpp
===================================================================
--- llvm-toolchain-snapshot_15~++20220314071859+3b61587c9e27.orig/lldb/source/Plugins/ScriptInterpreter/Python/ScriptInterpreterPython.cpp
+++ llvm-toolchain-snapshot_15~++20220314071859+3b61587c9e27/lldb/source/Plugins/ScriptInterpreter/Python/ScriptInterpreterPython.cpp
@@ -265,6 +265,8 @@ void ScriptInterpreterPython::ComputePyt
   // x86_64, or bin on Windows).
   llvm::sys::path::remove_filename(path);
   llvm::sys::path::append(path, LLDB_PYTHON_RELATIVE_LIBDIR);
+  // remove double "lib/lib" statement
+  std::replace(path.begin(), path.end(), '/lib/lib/', '/lib/');
 
 #if defined(_WIN32)
   // This will be injected directly through FileSpec.GetDirectory().SetString(),
