Index: llvm-toolchain-snapshot_3.8~svn246750/clang/tools/scan-build/scan-build
===================================================================
--- llvm-toolchain-snapshot_3.8~svn246750.orig/clang/tools/scan-build/scan-build
+++ llvm-toolchain-snapshot_3.8~svn246750/clang/tools/scan-build/scan-build
@@ -1642,7 +1642,7 @@ if (!@ARGV and !$RequestDisplayHelp) {
 
 # Find 'clang'
 if (!defined $Options{AnalyzerDiscoveryMethod}) {
-  $Clang = Cwd::realpath("$RealBin/bin/clang");
+  $Clang = Cwd::realpath("/usr/lib/llvm-3.8/bin/clang");
   if (!defined $Clang || ! -x $Clang) {
     $Clang = Cwd::realpath("$RealBin/clang");
   }
