Index: llvm-toolchain_3.3~svn176083/clang/tools/scan-build/scan-build
===================================================================
--- llvm-toolchain_3.3~svn176083.orig/clang/tools/scan-build/scan-build	2013-02-26 14:15:04.000000000 +0100
+++ llvm-toolchain_3.3~svn176083/clang/tools/scan-build/scan-build	2013-02-26 14:15:06.000000000 +0100
@@ -1512,7 +1512,7 @@
 
 # Find 'clang'
 if (!defined $AnalyzerDiscoveryMethod) {
-  $Clang = Cwd::realpath("$RealBin/bin/clang");
+  $Clang = Cwd::realpath("/usr/bin/clang");
   if (!defined $Clang || ! -x $Clang) {
     $Clang = Cwd::realpath("$RealBin/clang");
   }
