Index: llvm-toolchain-build/clang/tools/scan-build/scan-build
===================================================================
--- llvm-toolchain-build.orig/clang/tools/scan-build/scan-build	2013-03-15 09:41:33.000000000 +0100
+++ llvm-toolchain-build/clang/tools/scan-build/scan-build	2013-03-15 09:43:16.000000000 +0100
@@ -1514,7 +1514,7 @@
 
 # Find 'clang'
 if (!defined $AnalyzerDiscoveryMethod) {
-  $Clang = Cwd::realpath("$RealBin/bin/clang");
+  $Clang = Cwd::realpath("/usr/bin/clang");
   if (!defined $Clang || ! -x $Clang) {
     $Clang = Cwd::realpath("$RealBin/clang");
   }
