Index: llvm-toolchain-snapshot_3.8~svn249699/clang/tools/scan-build/scan-build
===================================================================
--- llvm-toolchain-snapshot_3.8~svn249699.orig/clang/tools/scan-build/scan-build
+++ llvm-toolchain-snapshot_3.8~svn249699/clang/tools/scan-build/scan-build
@@ -1405,7 +1405,7 @@ sub ShellEscape {
 
 sub FindClang {
   if (!defined $Options{AnalyzerDiscoveryMethod}) {
-    $Clang = Cwd::realpath("$RealBin/bin/clang") if (-f "$RealBin/bin/clang");
+    $Clang = Cwd::realpath("/usr/lib/llvm-3.8/bin/clang");
     if (!defined $Clang || ! -x $Clang) {
       $Clang = Cwd::realpath("$RealBin/clang") if (-f "$RealBin/clang");
     }
