Index: llvm-toolchain-snapshot_3.4~svn185325/clang/tools/scan-build/scan-build
===================================================================
--- llvm-toolchain-snapshot_3.4~svn185325.orig/clang/tools/scan-build/scan-build	2013-07-01 13:28:41.000000000 +0200
+++ llvm-toolchain-snapshot_3.4~svn185325/clang/tools/scan-build/scan-build	2013-07-01 13:28:41.000000000 +0200
@@ -1538,7 +1538,7 @@
 
 # Find 'clang'
 if (!defined $AnalyzerDiscoveryMethod) {
-  $Clang = Cwd::realpath("$RealBin/bin/clang");
+  $Clang = Cwd::realpath("/usr/bin/clang");
   if (!defined $Clang || ! -x $Clang) {
     $Clang = Cwd::realpath("$RealBin/clang");
   }
