Index: llvm-toolchain-snapshot_3.7~svn242297/autoconf/configure.ac
===================================================================
--- llvm-toolchain-snapshot_3.7~svn242297.orig/autoconf/configure.ac
+++ llvm-toolchain-snapshot_3.7~svn242297/autoconf/configure.ac
@@ -32,12 +32,12 @@ dnl===----------------------------------
 dnl Initialize autoconf and define the package name, version number and
 dnl address for reporting bugs.
 
-AC_INIT([LLVM],[3.8.0svn],[http://llvm.org/bugs/])
+AC_INIT([LLVM],[3.8.0],[http://llvm.org/bugs/])
 
 LLVM_VERSION_MAJOR=3
 LLVM_VERSION_MINOR=8
 LLVM_VERSION_PATCH=0
-LLVM_VERSION_SUFFIX=svn
+LLVM_VERSION_SUFFIX=
 
 AC_DEFINE_UNQUOTED([LLVM_VERSION_MAJOR], $LLVM_VERSION_MAJOR, [Major version of the LLVM API])
 AC_DEFINE_UNQUOTED([LLVM_VERSION_MINOR], $LLVM_VERSION_MINOR, [Minor version of the LLVM API])
Index: llvm-toolchain-snapshot_3.7~svn242297/configure
===================================================================
--- llvm-toolchain-snapshot_3.7~svn242297.orig/configure
+++ llvm-toolchain-snapshot_3.7~svn242297/configure
@@ -1956,7 +1956,7 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
 LLVM_VERSION_MAJOR=3
 LLVM_VERSION_MINOR=8
 LLVM_VERSION_PATCH=0
-LLVM_VERSION_SUFFIX=svn
+LLVM_VERSION_SUFFIX=
 
 
 cat >>confdefs.h <<_ACEOF
