Index: llvm-toolchain-13_13.0.0~+rc1/integration-test-suite/tests/test_tsan.c
===================================================================
--- llvm-toolchain-13_13.0.0~+rc1.orig/integration-test-suite/tests/test_tsan.c
+++ llvm-toolchain-13_13.0.0~+rc1/integration-test-suite/tests/test_tsan.c
@@ -5,7 +5,7 @@
 // RUN: %llvm-nm %t | grep __tsan
 // RUN: env TSAN_OPTIONS="log_path=stdout:exitcode=0"  %t 2>&1 > %t.out
 // RUN: grep -q "data race" %t.out
-// XFAIL: arm 
+// XFAIL: arm, i686, i386
 
 #include <pthread.h>
 #include <stdio.h>
