Index: llvm-toolchain-snapshot_3.5~svn203903/polly/lib/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.5~svn203903.orig/polly/lib/Makefile	2014-03-14 09:01:28.397124053 +0100
+++ llvm-toolchain-snapshot_3.5~svn203903/polly/lib/Makefile	2014-03-14 09:02:20.821790117 +0100
@@ -27,9 +27,6 @@
 DIRS += Support
 USEDLIBS += pollysupport.a
 
-DIRS += JSON
-USEDLIBS += pollyjson.a
-
 DIRS += Transform
 USEDLIBS += pollytransform.a
 
@@ -40,7 +37,7 @@
 #
 include $(LEVEL)/Makefile.common
 
-LIBS += $(POLLY_LD) $(POLLY_LIB)
+LIBS += $(POLLY_LD) $(POLLY_LIB) -ljsoncpp
 
 $(LibDir)/libpollyanalysis.a : $(LibDir)/.dir $(PROJ_OBJ_DIR)/Analysis/$(BuildMode)/.dir \
                                $(PROJ_SRC_DIR)/Analysis/*
@@ -106,4 +103,4 @@
       $(CP) $(PROJ_SRC_DIR)/Transform/Makefile Transform/Makefile; \
     fi; \
     ($(MAKE) -C Transform $@ ) || exit 1; \
-  fi
\ No newline at end of file
+  fi
