Index: llvm-toolchain-snapshot_3.6~svn214390/polly/lib/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.6~svn214390.orig/polly/lib/Makefile
+++ llvm-toolchain-snapshot_3.6~svn214390/polly/lib/Makefile
@@ -13,7 +13,7 @@ LOADABLE_MODULE = 1
 CPP.Flags += $(POLLY_INC)
 LD.Flags += $(POLLY_LD) $(POLLY_LIB)
 
-LIBS += $(POLLY_LD) $(POLLY_LIB)
+LIBS += $(POLLY_LD) $(POLLY_LIB) -ljsoncpp
 
 include $(LEVEL)/Makefile.config
 
@@ -42,9 +42,9 @@ ISL_CODEGEN_FILES= CodeGen/IslAst.cpp \
                    CodeGen/IslExprBuilder.cpp \
                    CodeGen/IslCodeGeneration.cpp
 
-POLLY_JSON_FILES= JSON/json_reader.cpp \
-		  JSON/json_value.cpp \
-		  JSON/json_writer.cpp
+# POLLY_JSON_FILES= JSON/json_reader.cpp \
+# 		  JSON/json_value.cpp \
+# 		  JSON/json_writer.cpp
 
 SOURCES= Polly.cpp \
 	 Support/GICHelper.cpp \
