Index: llvm-toolchain-snapshot_10~svn372737/utils/TableGen/CodeEmitterGen.cpp
===================================================================
--- llvm-toolchain-snapshot_10~svn372737.orig/utils/TableGen/CodeEmitterGen.cpp
+++ llvm-toolchain-snapshot_10~svn372737/utils/TableGen/CodeEmitterGen.cpp
@@ -395,6 +395,9 @@ void CodeEmitterGen::run(raw_ostream &o)
   ArrayRef<const CodeGenInstruction*> NumberedInstructions =
     Target.getInstructionsByEnumValue();
 
+  o << "// Undef for HURD\n";
+  o << "#ifdef EIEIO\n#undef EIEIO\n#endif\n";
+
   const CodeGenHwModes &HWM = Target.getHwModes();
   // The set of HwModes used by instruction encodings.
   std::set<unsigned> HwModes;
