Index: llvm-toolchain-snapshot_10~svn371937/utils/TableGen/CodeEmitterGen.cpp
===================================================================
--- llvm-toolchain-snapshot_10~svn371937.orig/utils/TableGen/CodeEmitterGen.cpp
+++ llvm-toolchain-snapshot_10~svn371937/utils/TableGen/CodeEmitterGen.cpp
@@ -320,6 +320,9 @@ void CodeEmitterGen::run(raw_ostream &o)
   ArrayRef<const CodeGenInstruction*> NumberedInstructions =
     Target.getInstructionsByEnumValue();
 
+  o << "// Undef for HURD\n";
+  o << "#ifdef EIEIO\n#undef EIEIO\n#endif\n";
+
   // Default to something sensible in case the target doesn't define Inst.
   BitWidth = 32;
   for (const CodeGenInstruction *CGI : NumberedInstructions) {
