Index: llvm-toolchain-snapshot_10~+20191020075550+5b8546023f3/llvm/utils/TableGen/CodeEmitterGen.cpp
===================================================================
--- llvm-toolchain-snapshot_10~+20191020075550+5b8546023f3.orig/llvm/utils/TableGen/CodeEmitterGen.cpp
+++ llvm-toolchain-snapshot_10~+20191020075550+5b8546023f3/llvm/utils/TableGen/CodeEmitterGen.cpp
@@ -396,6 +396,9 @@ void CodeEmitterGen::run(raw_ostream &o)
   ArrayRef<const CodeGenInstruction*> NumberedInstructions =
     Target.getInstructionsByEnumValue();
 
+  o << "// Undef for HURD\n";
+  o << "#ifdef EIEIO\n#undef EIEIO\n#endif\n";
+
   const CodeGenHwModes &HWM = Target.getHwModes();
   // The set of HwModes used by instruction encodings.
   std::set<unsigned> HwModes;
