Index: llvm-toolchain-snapshot_10~svn370025/llvm/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_10~svn370025.orig/llvm/CMakeLists.txt
+++ llvm-toolchain-snapshot_10~svn370025/llvm/CMakeLists.txt
@@ -866,7 +866,7 @@ if(LLVM_TARGET_IS_CROSSCOMPILE_HOST)
 # (this is a variable that CrossCompile sets on recursive invocations)
 endif()
 
-if(${CMAKE_SYSTEM_NAME} MATCHES "(FreeBSD|DragonFly)")
+if( ${CMAKE_SYSTEM_NAME} MATCHES "(FreeBSD|DragonFly)" AND NOT "kFreeBSD" )
   # On FreeBSD, /usr/local/* is not used by default. In order to build LLVM
   # with libxml2, iconv.h, etc., we must add /usr/local paths.
   include_directories(SYSTEM "/usr/local/include")
