Index: llvm-toolchain-snapshot_9~svn358740/include/llvm/ADT/Triple.h
===================================================================
--- llvm-toolchain-snapshot_9~svn358740.orig/include/llvm/ADT/Triple.h
+++ llvm-toolchain-snapshot_9~svn358740/include/llvm/ADT/Triple.h
@@ -159,7 +159,7 @@ public:
     FreeBSD,
     Fuchsia,
     IOS,
-    KFreeBSD,
+    kFreeBSD,
     Linux,
     Lv2,        // PS3
     MacOSX,
@@ -579,8 +579,8 @@ public:
   }
 
   /// Tests whether the OS is kFreeBSD.
-  bool isOSKFreeBSD() const {
-    return getOS() == Triple::KFreeBSD;
+  bool isOSkFreeBSD() const {
+    return getOS() == Triple::kFreeBSD;
   }
 
   /// Tests whether the OS is Hurd.
@@ -600,7 +600,7 @@ public:
 
   /// Tests whether the OS uses glibc.
   bool isOSGlibc() const {
-    return (getOS() == Triple::Linux || getOS() == Triple::KFreeBSD ||
+    return (getOS() == Triple::Linux || getOS() == Triple::kFreeBSD ||
             getOS() == Triple::Hurd) &&
            !isAndroid();
   }
