Index: llvm-toolchain-snapshot_9~svn353038/lib/Support/Triple.cpp
===================================================================
--- llvm-toolchain-snapshot_9~svn353038.orig/lib/Support/Triple.cpp
+++ llvm-toolchain-snapshot_9~svn353038/lib/Support/Triple.cpp
@@ -181,7 +181,7 @@ StringRef Triple::getOSTypeName(OSType K
   case FreeBSD: return "freebsd";
   case Fuchsia: return "fuchsia";
   case IOS: return "ios";
-  case KFreeBSD: return "kfreebsd";
+  case kFreeBSD: return "kfreebsd";
   case Linux: return "linux";
   case Lv2: return "lv2";
   case MacOSX: return "macosx";
@@ -478,7 +478,7 @@ static Triple::OSType parseOS(StringRef
     .StartsWith("freebsd", Triple::FreeBSD)
     .StartsWith("fuchsia", Triple::Fuchsia)
     .StartsWith("ios", Triple::IOS)
-    .StartsWith("kfreebsd", Triple::KFreeBSD)
+    .StartsWith("kfreebsd", Triple::kFreeBSD)
     .StartsWith("linux", Triple::Linux)
     .StartsWith("lv2", Triple::Lv2)
     .StartsWith("macos", Triple::MacOSX)
Index: llvm-toolchain-snapshot_9~svn353038/lib/Support/Unix/Path.inc
===================================================================
--- llvm-toolchain-snapshot_9~svn353038.orig/lib/Support/Unix/Path.inc
+++ llvm-toolchain-snapshot_9~svn353038/lib/Support/Unix/Path.inc
@@ -61,7 +61,7 @@
 #define FSTATVFS fstatvfs
 #define STATVFS_F_FRSIZE(vfs) vfs.f_frsize
 #else
-#if defined(__OpenBSD__) || defined(__FreeBSD__)
+#if defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 #include <sys/mount.h>
 #include <sys/param.h>
 #elif defined(__linux__)
